[2024-08-14 11:31:57,533: INFO: main: Welcome to our custom logger:]
[2024-08-14 23:35:49,022: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-14 23:35:49,029: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-14 23:35:49,035: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-14 23:35:49,035: ERROR: main: name 'BoxvalueError' is not defined:]
Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/utils/common.py", line 26, in read_yaml
    return ConfigBox(content)
           ^^^^^^^^^^^^^^^^^^
  File "box/box.py", line 284, in box.box.Box.__init__
box.exceptions.BoxValueError: Cannot extrapolate Box from string

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/main.py", line 9, in <module>
    obj.main()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/pipeline/stage_01_data_ingestion.py", line 13, in main
    config = ConfigurationManager()
             ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/config/configuration.py", line 14, in __init__
    self.params = read_yaml(params_filepath)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/.local/pip3/venv/lib/python3.12/site-packages/ensure/main.py", line 872, in __call__
    return_val = self.f(*args, **kwargs)
                 ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/utils/common.py", line 28, in read_yaml
    except BoxvalueError:
           ^^^^^^^^^^^^^
NameError: name 'BoxvalueError' is not defined. Did you mean: 'BoxValueError'?
[2024-08-14 23:36:41,149: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-14 23:36:41,153: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-14 23:36:41,159: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-14 23:36:41,160: ERROR: main: Yaml file is empty:]
Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/utils/common.py", line 26, in read_yaml
    return ConfigBox(content)
           ^^^^^^^^^^^^^^^^^^
  File "box/box.py", line 284, in box.box.Box.__init__
box.exceptions.BoxValueError: Cannot extrapolate Box from string

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/main.py", line 9, in <module>
    obj.main()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/pipeline/stage_01_data_ingestion.py", line 13, in main
    config = ConfigurationManager()
             ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/config/configuration.py", line 14, in __init__
    self.params = read_yaml(params_filepath)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/.local/pip3/venv/lib/python3.12/site-packages/ensure/main.py", line 872, in __call__
    return_val = self.f(*args, **kwargs)
                 ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/utils/common.py", line 29, in read_yaml
    raise ValueError("Yaml file is empty")
ValueError: Yaml file is empty
[2024-08-14 23:57:37,940: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-14 23:57:37,945: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-14 23:57:37,964: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-14 23:57:37,965: ERROR: main: Yaml file is empty:]
Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/utils/common.py", line 26, in read_yaml
    return ConfigBox(content)
           ^^^^^^^^^^^^^^^^^^
  File "box/box.py", line 284, in box.box.Box.__init__
box.exceptions.BoxValueError: Cannot extrapolate Box from string

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/main.py", line 9, in <module>
    obj.main()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/pipeline/stage_01_data_ingestion.py", line 13, in main
    config = ConfigurationManager()
             ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/config/configuration.py", line 14, in __init__
    self.params = read_yaml(params_filepath)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/.local/pip3/venv/lib/python3.12/site-packages/ensure/main.py", line 872, in __call__
    return_val = self.f(*args, **kwargs)
                 ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/utils/common.py", line 29, in read_yaml
    raise ValueError("Yaml file is empty")
ValueError: Yaml file is empty
[2024-08-14 23:59:55,176: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-14 23:59:55,180: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-14 23:59:55,192: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-14 23:59:55,192: ERROR: main: Cannot extrapolate Box from string:]
Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/main.py", line 9, in <module>
    obj.main()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/pipeline/stage_01_data_ingestion.py", line 13, in main
    config = ConfigurationManager()
             ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/config/configuration.py", line 14, in __init__
    self.params = read_yaml(params_filepath)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/.local/pip3/venv/lib/python3.12/site-packages/ensure/main.py", line 872, in __call__
    return_val = self.f(*args, **kwargs)
                 ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/utils/common.py", line 31, in read_yaml
    raise e
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/utils/common.py", line 26, in read_yaml
    return ConfigBox(content)
           ^^^^^^^^^^^^^^^^^^
  File "box/box.py", line 284, in box.box.Box.__init__
box.exceptions.BoxValueError: Cannot extrapolate Box from string
[2024-08-15 00:00:23,043: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 00:00:23,047: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 00:00:23,064: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 00:00:23,064: ERROR: main: Cannot extrapolate Box from string:]
Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/main.py", line 9, in <module>
    obj.main()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/pipeline/stage_01_data_ingestion.py", line 13, in main
    config = ConfigurationManager()
             ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/config/configuration.py", line 14, in __init__
    self.params = read_yaml(params_filepath)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/.local/pip3/venv/lib/python3.12/site-packages/ensure/main.py", line 872, in __call__
    return_val = self.f(*args, **kwargs)
                 ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/utils/common.py", line 31, in read_yaml
    raise e
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/utils/common.py", line 26, in read_yaml
    return ConfigBox(content)
           ^^^^^^^^^^^^^^^^^^
  File "box/box.py", line 284, in box.box.Box.__init__
box.exceptions.BoxValueError: Cannot extrapolate Box from string
[2024-08-15 00:05:36,316: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 00:05:36,320: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 00:05:36,326: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 00:05:36,332: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 00:05:36,335: ERROR: main: create_directories() missing 1 required positional argument: 'verbose':]
Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/main.py", line 9, in <module>
    obj.main()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/pipeline/stage_01_data_ingestion.py", line 13, in main
    config = ConfigurationManager()
             ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/config/configuration.py", line 17, in __init__
    create_directories([self.config.artifacts_root])
  File "/Users/deepa/.local/pip3/venv/lib/python3.12/site-packages/ensure/main.py", line 835, in __call__
    return self.f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^
TypeError: create_directories() missing 1 required positional argument: 'verbose'
[2024-08-15 00:08:00,432: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 00:08:00,437: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 00:08:00,447: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 00:08:00,455: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 00:08:00,458: ERROR: main: isinstance() arg 2 must be a type, a tuple of types, or a union:]
Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/main.py", line 9, in <module>
    obj.main()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/pipeline/stage_01_data_ingestion.py", line 13, in main
    config = ConfigurationManager()
             ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/config/configuration.py", line 17, in __init__
    create_directories([self.config.artifacts_root], True)
  File "/Users/deepa/.local/pip3/venv/lib/python3.12/site-packages/ensure/main.py", line 831, in __call__
    if not isinstance(value, templ):
           ^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: isinstance() arg 2 must be a type, a tuple of types, or a union
[2024-08-15 00:10:14,476: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 00:10:14,481: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 00:10:14,486: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 00:10:14,489: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 00:10:14,495: ERROR: main: create_directories() missing 1 required positional argument: 'verbose':]
Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/main.py", line 9, in <module>
    obj.main()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/pipeline/stage_01_data_ingestion.py", line 13, in main
    config = ConfigurationManager()
             ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/config/configuration.py", line 17, in __init__
    create_directories([self.config.artifacts_root])
  File "/Users/deepa/.local/pip3/venv/lib/python3.12/site-packages/ensure/main.py", line 835, in __call__
    return self.f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^
TypeError: create_directories() missing 1 required positional argument: 'verbose'
[2024-08-15 00:11:23,536: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 00:11:23,540: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 00:11:23,545: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 00:11:23,548: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 00:11:23,550: INFO: common: created directory at artifacts:]
[2024-08-15 00:11:23,551: ERROR: main: "'ConfigBox' object has no attribute 'root_dir'":]
Traceback (most recent call last):
  File "box/box.py", line 593, in box.box.Box.__getitem__
KeyError: 'root_dir'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "box/box.py", line 632, in box.box.Box.__getattr__
  File "box/box.py", line 620, in box.box.Box.__getitem__
box.exceptions.BoxKeyError: "'root_dir'"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "box/box.py", line 634, in box.box.Box.__getattr__
AttributeError: 'ConfigBox' object has no attribute 'root_dir'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "box/config_box.py", line 29, in box.config_box.ConfigBox.__getattr__
  File "box/box.py", line 648, in box.box.Box.__getattr__
box.exceptions.BoxKeyError: "'ConfigBox' object has no attribute 'root_dir'"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "box/box.py", line 593, in box.box.Box.__getitem__
KeyError: 'root_dir'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "box/box.py", line 632, in box.box.Box.__getattr__
  File "box/box.py", line 620, in box.box.Box.__getitem__
box.exceptions.BoxKeyError: "'root_dir'"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "box/box.py", line 634, in box.box.Box.__getattr__
AttributeError: 'ConfigBox' object has no attribute 'root_dir'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/main.py", line 9, in <module>
    obj.main()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/pipeline/stage_01_data_ingestion.py", line 14, in main
    data_ingestion_config = config.get_data_ingestion_config()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/config/configuration.py", line 24, in get_data_ingestion_config
    create_directories([self.config.root_dir])
                        ^^^^^^^^^^^^^^^^^^^^
  File "box/config_box.py", line 31, in box.config_box.ConfigBox.__getattr__
  File "box/box.py", line 648, in box.box.Box.__getattr__
box.exceptions.BoxKeyError: "'ConfigBox' object has no attribute 'root_dir'"
[2024-08-15 00:17:37,505: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 00:17:37,509: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 00:17:37,515: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 00:17:37,539: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 00:17:37,544: INFO: common: created directory at artifacts:]
[2024-08-15 00:17:37,546: ERROR: main: "'ConfigBox' object has no attribute 'root_dir'":]
Traceback (most recent call last):
  File "box/box.py", line 593, in box.box.Box.__getitem__
KeyError: 'root_dir'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "box/box.py", line 632, in box.box.Box.__getattr__
  File "box/box.py", line 620, in box.box.Box.__getitem__
box.exceptions.BoxKeyError: "'root_dir'"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "box/box.py", line 634, in box.box.Box.__getattr__
AttributeError: 'ConfigBox' object has no attribute 'root_dir'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "box/config_box.py", line 29, in box.config_box.ConfigBox.__getattr__
  File "box/box.py", line 648, in box.box.Box.__getattr__
box.exceptions.BoxKeyError: "'ConfigBox' object has no attribute 'root_dir'"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "box/box.py", line 593, in box.box.Box.__getitem__
KeyError: 'root_dir'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "box/box.py", line 632, in box.box.Box.__getattr__
  File "box/box.py", line 620, in box.box.Box.__getitem__
box.exceptions.BoxKeyError: "'root_dir'"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "box/box.py", line 634, in box.box.Box.__getattr__
AttributeError: 'ConfigBox' object has no attribute 'root_dir'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/main.py", line 9, in <module>
    obj.main()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/pipeline/stage_01_data_ingestion.py", line 14, in main
    data_ingestion_config = config.get_data_ingestion_config()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/config/configuration.py", line 24, in get_data_ingestion_config
    create_directories([self.config.root_dir])
                        ^^^^^^^^^^^^^^^^^^^^
  File "box/config_box.py", line 31, in box.config_box.ConfigBox.__getattr__
  File "box/box.py", line 648, in box.box.Box.__getattr__
box.exceptions.BoxKeyError: "'ConfigBox' object has no attribute 'root_dir'"
[2024-08-15 09:23:20,241: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 09:23:20,244: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 09:23:20,251: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 09:23:20,260: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 09:23:20,277: INFO: common: created directory at artifacts:]
[2024-08-15 09:23:20,277: ERROR: main: create_directories() missing 1 required positional argument: 'verbose':]
Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/main.py", line 9, in <module>
    obj.main()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/pipeline/stage_01_data_ingestion.py", line 14, in main
    data_ingestion_config = config.get_data_ingestion_config()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/config/configuration.py", line 24, in get_data_ingestion_config
    create_directories([config.root_dir])
  File "/Users/deepa/.local/pip3/venv/lib/python3.12/site-packages/ensure/main.py", line 835, in __call__
    return self.f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^
TypeError: create_directories() missing 1 required positional argument: 'verbose'
[2024-08-15 09:24:43,595: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 09:24:43,600: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 09:24:43,607: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 09:24:43,611: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 09:24:43,613: ERROR: main: create_directories() missing 1 required positional argument: 'verbose':]
Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/main.py", line 9, in <module>
    obj.main()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/pipeline/stage_01_data_ingestion.py", line 13, in main
    config = ConfigurationManager()
             ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/config/configuration.py", line 17, in __init__
    create_directories([self.config.artifacts_root])
  File "/Users/deepa/.local/pip3/venv/lib/python3.12/site-packages/ensure/main.py", line 835, in __call__
    return self.f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^
TypeError: create_directories() missing 1 required positional argument: 'verbose'
[2024-08-15 09:28:51,769: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 09:28:51,773: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 09:28:51,779: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 09:28:51,784: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 09:28:51,787: INFO: common: created directory at artifacts:]
[2024-08-15 09:28:51,789: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 09:28:51,789: ERROR: main: "'ConfigBox' object has no attribute 'source_url'":]
Traceback (most recent call last):
  File "box/box.py", line 593, in box.box.Box.__getitem__
KeyError: 'Source_URL'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "box/box.py", line 632, in box.box.Box.__getattr__
  File "box/box.py", line 620, in box.box.Box.__getitem__
box.exceptions.BoxKeyError: "'Source_URL'"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "box/box.py", line 634, in box.box.Box.__getattr__
AttributeError: 'ConfigBox' object has no attribute 'Source_URL'. Did you mean: 'source_URL'?

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "box/config_box.py", line 29, in box.config_box.ConfigBox.__getattr__
  File "box/box.py", line 648, in box.box.Box.__getattr__
box.exceptions.BoxKeyError: "'ConfigBox' object has no attribute 'Source_URL'"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "box/box.py", line 593, in box.box.Box.__getitem__
KeyError: 'source_url'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "box/box.py", line 632, in box.box.Box.__getattr__
  File "box/box.py", line 620, in box.box.Box.__getitem__
box.exceptions.BoxKeyError: "'source_url'"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "box/box.py", line 634, in box.box.Box.__getattr__
AttributeError: 'ConfigBox' object has no attribute 'source_url'. Did you mean: 'source_URL'?

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/main.py", line 9, in <module>
    obj.main()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/pipeline/stage_01_data_ingestion.py", line 14, in main
    data_ingestion_config = config.get_data_ingestion_config()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/config/configuration.py", line 28, in get_data_ingestion_config
    source_URL = config.Source_URL,
                 ^^^^^^^^^^^^^^^^^
  File "box/config_box.py", line 31, in box.config_box.ConfigBox.__getattr__
  File "box/box.py", line 648, in box.box.Box.__getattr__
box.exceptions.BoxKeyError: "'ConfigBox' object has no attribute 'source_url'". Did you mean: 'source_URL'?
[2024-08-15 09:30:37,286: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 09:30:37,290: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 09:30:37,296: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 09:30:37,300: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 09:30:37,305: INFO: common: created directory at artifacts:]
[2024-08-15 09:30:37,306: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 09:30:38,508: ERROR: main: [Errno 2] No such file or directory: 'artifacts/data_injestion/data.zip':]
Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/main.py", line 9, in <module>
    obj.main()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/pipeline/stage_01_data_ingestion.py", line 16, in main
    data_ingestion.download_file()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/components/data_ingestion.py", line 16, in download_file
    filename, headers = request.urlretrieve(url=self.config.source_URL,
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/Cellar/python@3.12/3.12.5/Frameworks/Python.framework/Versions/3.12/lib/python3.12/urllib/request.py", line 250, in urlretrieve
    tfp = open(filename, 'wb')
          ^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: 'artifacts/data_injestion/data.zip'
[2024-08-15 09:35:26,483: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 09:35:26,487: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 09:35:26,499: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 09:35:26,508: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 09:35:26,510: INFO: common: created directory at artifacts:]
[2024-08-15 09:35:26,510: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 09:35:27,073: ERROR: main: [Errno 2] No such file or directory: 'artifacts/data_injestion/data.zip':]
Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/main.py", line 9, in <module>
    obj.main()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/pipeline/stage_01_data_ingestion.py", line 16, in main
    data_ingestion.download_file()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/components/data_ingestion.py", line 16, in download_file
    filename, headers = request.urlretrieve(url=self.config.source_URL,
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/Cellar/python@3.12/3.12.5/Frameworks/Python.framework/Versions/3.12/lib/python3.12/urllib/request.py", line 250, in urlretrieve
    tfp = open(filename, 'wb')
          ^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: 'artifacts/data_injestion/data.zip'
[2024-08-15 09:36:26,960: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 09:36:26,963: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 09:36:26,969: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 09:36:26,976: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 09:36:26,979: INFO: common: created directory at artifacts:]
[2024-08-15 09:36:26,979: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 09:36:27,490: ERROR: main: [Errno 2] No such file or directory: 'artifacts/data_injestion/winequality-data.zip':]
Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/main.py", line 9, in <module>
    obj.main()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/pipeline/stage_01_data_ingestion.py", line 16, in main
    data_ingestion.download_file()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/components/data_ingestion.py", line 16, in download_file
    filename, headers = request.urlretrieve(url=self.config.source_URL,
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/Cellar/python@3.12/3.12.5/Frameworks/Python.framework/Versions/3.12/lib/python3.12/urllib/request.py", line 250, in urlretrieve
    tfp = open(filename, 'wb')
          ^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: 'artifacts/data_injestion/winequality-data.zip'
[2024-08-15 09:39:33,024: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 09:39:33,027: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 09:39:33,033: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 09:39:33,038: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 09:39:33,046: INFO: common: created directory at artifacts:]
[2024-08-15 09:39:33,046: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 09:39:33,569: INFO: data_ingestion: artifacts/data_ingestion/data.zip download with following headers Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 4A90:631B5:2422A1:264D4A:66BE2D2E
Accept-Ranges: bytes
Date: Thu, 15 Aug 2024 16:39:34 GMT
Via: 1.1 varnish
X-Served-By: cache-pao-kpao1770080-PAO
X-Cache: HIT
X-Cache-Hits: 0
X-Timer: S1723739974.233187,VS0,VE1
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: 18be0de0121a0d33041e3410dc56ae7b488cb76d
Expires: Thu, 15 Aug 2024 16:44:34 GMT
Source-Age: 186

:]
[2024-08-15 09:39:33,576: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has completed

**********:]
[2024-08-15 09:43:34,521: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 09:43:34,525: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 09:43:34,549: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 09:43:34,551: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 09:43:34,553: INFO: common: created directory at artifacts:]
[2024-08-15 09:43:34,553: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 09:43:35,634: INFO: data_ingestion: artifacts/data_ingestion/data.zip download with following headers Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 4A90:631B5:2422A1:264D4A:66BE2D2E
Accept-Ranges: bytes
Date: Thu, 15 Aug 2024 16:43:36 GMT
Via: 1.1 varnish
X-Served-By: cache-pao-kpao1770068-PAO
X-Cache: HIT
X-Cache-Hits: 0
X-Timer: S1723740216.062372,VS0,VE141
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: 437aff4a1a30a8c8bdb7f2c2a44c0687dd159ef2
Expires: Thu, 15 Aug 2024 16:48:36 GMT
Source-Age: 0

:]
[2024-08-15 09:43:35,640: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has completed

**********:]
[2024-08-15 09:43:35,640: INFO: main: starting <<<<<<<<< {stage_name} >>>>>>>>>>>:]
[2024-08-15 09:43:35,644: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 09:43:35,646: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 09:43:35,659: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 09:43:35,659: INFO: common: created directory at artifacts:]
[2024-08-15 09:43:35,660: INFO: common: created directory at artifacts/data_validation:]
[2024-08-15 09:43:35,710: INFO: main:  <<<<<<<<< {stage_name} completed >>>>>>>>>>>:]
[2024-08-15 09:44:28,852: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 09:44:28,855: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 09:44:28,868: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 09:44:28,871: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 09:44:28,873: INFO: common: created directory at artifacts:]
[2024-08-15 09:44:28,873: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 09:44:29,351: INFO: data_ingestion: artifacts/data_ingestion/data.zip download with following headers Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 4A90:631B5:2422A1:264D4A:66BE2D2E
Accept-Ranges: bytes
Date: Thu, 15 Aug 2024 16:44:30 GMT
Via: 1.1 varnish
X-Served-By: cache-pao-kpao1770028-PAO
X-Cache: HIT
X-Cache-Hits: 1
X-Timer: S1723740270.017442,VS0,VE1
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: 8c3217a754286a237e2d88be24f175fd45162694
Expires: Thu, 15 Aug 2024 16:49:30 GMT
Source-Age: 54

:]
[2024-08-15 09:44:29,357: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has completed

**********:]
[2024-08-15 09:44:29,357: INFO: main: starting <<<<<<<<< Data validation Stage >>>>>>>>>>>:]
[2024-08-15 09:44:29,366: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 09:44:29,367: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 09:44:29,369: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 09:44:29,370: INFO: common: created directory at artifacts:]
[2024-08-15 09:44:29,370: INFO: common: created directory at artifacts/data_validation:]
[2024-08-15 09:44:29,394: INFO: main:  <<<<<<<<< Data validation Stage completed >>>>>>>>>>>:]
[2024-08-15 10:27:12,259: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 10:27:12,263: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 10:27:12,270: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 10:27:12,281: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 10:27:12,282: INFO: common: created directory at artifacts:]
[2024-08-15 10:27:12,283: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 10:27:13,175: INFO: data_ingestion: artifacts/data_ingestion/data.zip download with following headers Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 4A90:631B5:2422A1:264D4A:66BE2D2E
Accept-Ranges: bytes
Date: Thu, 15 Aug 2024 17:27:13 GMT
Via: 1.1 varnish
X-Served-By: cache-pao-kpao1770054-PAO
X-Cache: HIT
X-Cache-Hits: 0
X-Timer: S1723742834.757388,VS0,VE131
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: a1a014c19548893865a8d56053c3a1d86bd1520b
Expires: Thu, 15 Aug 2024 17:32:13 GMT
Source-Age: 0

:]
[2024-08-15 10:27:13,183: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has completed

**********:]
[2024-08-15 10:27:13,183: INFO: main: starting <<<<<<<<< Data validation Stage >>>>>>>>>>>:]
[2024-08-15 10:27:13,188: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 10:27:13,189: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 10:27:13,194: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 10:27:13,195: INFO: common: created directory at artifacts:]
[2024-08-15 10:27:13,196: INFO: common: created directory at artifacts/data_validation:]
[2024-08-15 10:27:13,221: INFO: main:  <<<<<<<<< Data validation Stage completed >>>>>>>>>>>:]
[2024-08-15 10:27:13,221: INFO: main: <<<<<<<<<< Data Transformation stage started >>>>>>>>>>>>>:]
[2024-08-15 10:27:13,224: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 10:27:13,226: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 10:27:13,232: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 10:27:13,233: INFO: common: created directory at artifacts:]
[2024-08-15 10:27:13,233: INFO: common: created directory at artifacts/data_transformation:]
[2024-08-15 10:27:13,233: ERROR: main: DataTransformationConfig.__init__() got an unexpected keyword argument 'root_dir':]
Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/main.py", line 35, in <module>
    obj.main()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/pipeline/stage_03_data_transformation.py", line 14, in main
    data_transformation_config = config.get_data_transformation_config()
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/config/configuration.py", line 53, in get_data_transformation_config
    data_transformation_config = DataTransformationConfig(
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DataTransformationConfig.__init__() got an unexpected keyword argument 'root_dir'
[2024-08-15 10:34:38,953: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 10:34:38,958: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 10:34:38,963: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 10:34:38,967: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 10:34:38,968: INFO: common: created directory at artifacts:]
[2024-08-15 10:34:38,972: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 10:34:39,688: INFO: data_ingestion: artifacts/data_ingestion/data.zip download with following headers Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 4A90:631B5:2422A1:264D4A:66BE2D2E
Accept-Ranges: bytes
Date: Thu, 15 Aug 2024 17:34:40 GMT
Via: 1.1 varnish
X-Served-By: cache-pao-kpao1770066-PAO
X-Cache: HIT
X-Cache-Hits: 0
X-Timer: S1723743280.293652,VS0,VE113
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: e3be98c414a9e5498b6c9fee68791104e9d91117
Expires: Thu, 15 Aug 2024 17:39:40 GMT
Source-Age: 0

:]
[2024-08-15 10:34:39,695: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has completed

**********:]
[2024-08-15 10:34:39,696: INFO: main: starting <<<<<<<<< Data validation Stage >>>>>>>>>>>:]
[2024-08-15 10:34:39,700: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 10:34:39,702: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 10:34:39,705: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 10:34:39,706: INFO: common: created directory at artifacts:]
[2024-08-15 10:34:39,707: INFO: common: created directory at artifacts/data_validation:]
[2024-08-15 10:34:39,728: INFO: main:  <<<<<<<<< Data validation Stage completed >>>>>>>>>>>:]
[2024-08-15 10:34:39,728: INFO: main: <<<<<<<<<< Data Transformation stage started >>>>>>>>>>>>>:]
[2024-08-15 10:34:39,731: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 10:34:39,732: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 10:34:39,737: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 10:34:39,738: INFO: common: created directory at artifacts:]
[2024-08-15 10:34:39,739: INFO: common: created directory at artifacts/data_transformation:]
[2024-08-15 10:34:39,739: ERROR: main: DataTransformationConfig.__init__() got an unexpected keyword argument 'root_dir':]
Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/main.py", line 35, in <module>
    obj.main()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/pipeline/stage_03_data_transformation.py", line 14, in main
    data_transformation_config = config.get_data_transformation_config()
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/config/configuration.py", line 53, in get_data_transformation_config
    data_transformation_config = DataTransformationConfig(
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DataTransformationConfig.__init__() got an unexpected keyword argument 'root_dir'
[2024-08-15 10:38:02,147: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 10:38:02,151: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 10:38:02,165: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 10:38:02,191: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 10:38:02,200: INFO: common: created directory at artifacts:]
[2024-08-15 10:38:02,201: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 10:38:02,778: INFO: data_ingestion: artifacts/data_ingestion/data.zip download with following headers Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 4A90:631B5:2422A1:264D4A:66BE2D2E
Accept-Ranges: bytes
Date: Thu, 15 Aug 2024 17:38:03 GMT
Via: 1.1 varnish
X-Served-By: cache-pao-kpao1770036-PAO
X-Cache: HIT
X-Cache-Hits: 0
X-Timer: S1723743483.490855,VS0,VE1
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: e7b22db7808fbd42d1a663eefa2a6d502d046d0f
Expires: Thu, 15 Aug 2024 17:43:03 GMT
Source-Age: 203

:]
[2024-08-15 10:38:02,785: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has completed

**********:]
[2024-08-15 10:38:02,786: INFO: main: starting <<<<<<<<< Data validation Stage >>>>>>>>>>>:]
[2024-08-15 10:38:02,792: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 10:38:02,794: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 10:38:02,798: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 10:38:02,800: INFO: common: created directory at artifacts:]
[2024-08-15 10:38:02,801: INFO: common: created directory at artifacts/data_validation:]
[2024-08-15 10:38:02,865: INFO: main:  <<<<<<<<< Data validation Stage completed >>>>>>>>>>>:]
[2024-08-15 10:38:02,865: INFO: main: <<<<<<<<<< Data Transformation stage started >>>>>>>>>>>>>:]
[2024-08-15 10:38:02,872: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 10:38:02,877: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 10:38:02,883: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 10:38:02,888: INFO: common: created directory at artifacts:]
[2024-08-15 10:38:02,889: INFO: common: created directory at artifacts/data_transformation:]
[2024-08-15 10:38:02,889: ERROR: main: DataTransformationConfig.__init__() got an unexpected keyword argument 'root_dir':]
Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/main.py", line 35, in <module>
    obj.main()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/pipeline/stage_03_data_transformation.py", line 14, in main
    data_transformation_config = config.get_data_transformation_config()
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/config/configuration.py", line 53, in get_data_transformation_config
    data_transformation_config = DataTransformationConfig(
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DataTransformationConfig.__init__() got an unexpected keyword argument 'root_dir'
[2024-08-15 10:58:13,458: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 10:58:13,463: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 10:58:13,469: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 10:58:13,473: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 10:58:13,476: INFO: common: created directory at artifacts:]
[2024-08-15 10:58:13,477: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 10:58:14,444: INFO: data_ingestion: artifacts/data_ingestion/data.zip download with following headers Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 4A90:631B5:2422A1:264D4A:66BE2D2E
Accept-Ranges: bytes
Date: Thu, 15 Aug 2024 17:58:15 GMT
Via: 1.1 varnish
X-Served-By: cache-pao-kpao1770051-PAO
X-Cache: HIT
X-Cache-Hits: 0
X-Timer: S1723744695.898173,VS0,VE127
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: 7d575e6a6848531a23e4283f3e873610d79a508a
Expires: Thu, 15 Aug 2024 18:03:15 GMT
Source-Age: 0

:]
[2024-08-15 10:58:14,452: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has completed

**********:]
[2024-08-15 10:58:14,453: INFO: main: starting <<<<<<<<< Data validation Stage >>>>>>>>>>>:]
[2024-08-15 10:58:14,462: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 10:58:14,466: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 10:58:14,470: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 10:58:14,475: INFO: common: created directory at artifacts:]
[2024-08-15 10:58:14,476: INFO: common: created directory at artifacts/data_validation:]
[2024-08-15 10:58:14,508: INFO: main:  <<<<<<<<< Data validation Stage completed >>>>>>>>>>>:]
[2024-08-15 10:58:14,508: INFO: main: <<<<<<<<<< Data Transformation stage started >>>>>>>>>>>>>:]
[2024-08-15 10:58:14,511: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 10:58:14,512: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 10:58:14,515: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 10:58:14,516: INFO: common: created directory at artifacts:]
[2024-08-15 10:58:14,517: INFO: common: created directory at artifacts/data_transformation:]
[2024-08-15 10:58:14,517: ERROR: main: DataTransformationConfig.__init__() got an unexpected keyword argument 'root_dir':]
Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/main.py", line 35, in <module>
    obj.main()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/pipeline/stage_03_data_transformation.py", line 14, in main
    data_transformation_config = config.get_data_transformation_config()
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/config/configuration.py", line 53, in get_data_transformation_config
    data_transformation_config = DataTransformationConfig(
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DataTransformationConfig.__init__() got an unexpected keyword argument 'root_dir'
[2024-08-15 10:59:45,878: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 10:59:45,883: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 10:59:45,889: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 10:59:45,895: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 10:59:45,903: INFO: common: created directory at artifacts:]
[2024-08-15 10:59:45,904: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 10:59:46,448: INFO: data_ingestion: artifacts/data_ingestion/data.zip download with following headers Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 4A90:631B5:2422A1:264D4A:66BE2D2E
Accept-Ranges: bytes
Date: Thu, 15 Aug 2024 17:59:48 GMT
Via: 1.1 varnish
X-Served-By: cache-pao-kpao1770040-PAO
X-Cache: HIT
X-Cache-Hits: 0
X-Timer: S1723744788.455643,VS0,VE1
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: a101e759dafd1e9f7dba82537e121e6bebe7597d
Expires: Thu, 15 Aug 2024 18:04:48 GMT
Source-Age: 93

:]
[2024-08-15 10:59:46,452: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has completed

**********:]
[2024-08-15 10:59:46,453: INFO: main: starting <<<<<<<<< Data validation Stage >>>>>>>>>>>:]
[2024-08-15 10:59:46,457: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 10:59:46,458: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 10:59:46,461: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 10:59:46,465: INFO: common: created directory at artifacts:]
[2024-08-15 10:59:46,469: INFO: common: created directory at artifacts/data_validation:]
[2024-08-15 10:59:46,530: INFO: main:  <<<<<<<<< Data validation Stage completed >>>>>>>>>>>:]
[2024-08-15 10:59:46,533: INFO: main: <<<<<<<<<< Data Transformation stage started >>>>>>>>>>>>>:]
[2024-08-15 10:59:46,537: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 10:59:46,540: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 10:59:46,545: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 10:59:46,547: INFO: common: created directory at artifacts:]
[2024-08-15 10:59:46,548: INFO: common: created directory at artifacts/data_transformation:]
[2024-08-15 10:59:46,548: ERROR: main: DataTransformationConfig.__init__() got an unexpected keyword argument 'root_dir':]
Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/main.py", line 35, in <module>
    obj.main()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/pipeline/stage_03_data_transformation.py", line 14, in main
    data_transformation_config = config.get_data_transformation_config()
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/config/configuration.py", line 53, in get_data_transformation_config
    data_transformation_config = DataTransformationConfig(
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DataTransformationConfig.__init__() got an unexpected keyword argument 'root_dir'
[2024-08-15 11:07:24,715: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 11:07:24,721: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 11:07:24,728: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 11:07:24,732: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 11:07:24,733: INFO: common: created directory at artifacts:]
[2024-08-15 11:07:24,733: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 11:07:25,616: INFO: data_ingestion: artifacts/data_ingestion/data.zip download with following headers Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 4A90:631B5:2422A1:264D4A:66BE2D2E
Accept-Ranges: bytes
Date: Thu, 15 Aug 2024 18:07:27 GMT
Via: 1.1 varnish
X-Served-By: cache-pao-kpao1770067-PAO
X-Cache: HIT
X-Cache-Hits: 0
X-Timer: S1723745247.378995,VS0,VE143
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: 62fce3d2f82dec368ce050bf0aff5c7654b15da3
Expires: Thu, 15 Aug 2024 18:12:27 GMT
Source-Age: 0

:]
[2024-08-15 11:07:25,621: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has completed

**********:]
[2024-08-15 11:07:25,621: INFO: main: starting <<<<<<<<< Data validation Stage >>>>>>>>>>>:]
[2024-08-15 11:07:25,630: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 11:07:25,635: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 11:07:25,649: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 11:07:25,650: INFO: common: created directory at artifacts:]
[2024-08-15 11:07:25,650: INFO: common: created directory at artifacts/data_validation:]
[2024-08-15 11:07:25,695: INFO: main:  <<<<<<<<< Data validation Stage completed >>>>>>>>>>>:]
[2024-08-15 11:07:25,695: INFO: main: <<<<<<<<<< Data Transformation stage started >>>>>>>>>>>>>:]
[2024-08-15 11:07:25,698: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 11:07:25,698: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 11:07:25,700: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 11:07:25,702: INFO: common: created directory at artifacts:]
[2024-08-15 11:07:25,703: INFO: common: created directory at artifacts/data_transformation:]
[2024-08-15 11:07:25,757: INFO: data_transformation: splitting data into train and test data:]
[2024-08-15 11:07:25,758: INFO: data_transformation: (1199, 12):]
[2024-08-15 11:07:25,758: INFO: data_transformation: (400, 12):]
[2024-08-15 11:07:25,758: INFO: main: <<<<<<<<<< Data Transformation stage completed >>>>>>>>>>>>>:]
[2024-08-15 11:09:55,408: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 11:09:55,412: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 11:09:55,419: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 11:09:55,423: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 11:09:55,426: INFO: common: created directory at artifacts:]
[2024-08-15 11:09:55,427: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 11:09:56,270: INFO: data_ingestion: artifacts/data_ingestion/data.zip download with following headers Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 4A90:631B5:2422A1:264D4A:66BE2D2E
Accept-Ranges: bytes
Date: Thu, 15 Aug 2024 18:09:58 GMT
Via: 1.1 varnish
X-Served-By: cache-pao-kpao1770069-PAO
X-Cache: HIT
X-Cache-Hits: 0
X-Timer: S1723745398.296061,VS0,VE1
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: 436e2e4b47c523000f32f6e81d320c1281b06785
Expires: Thu, 15 Aug 2024 18:14:58 GMT
Source-Age: 151

:]
[2024-08-15 11:09:56,274: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has completed

**********:]
[2024-08-15 11:09:56,275: INFO: main: starting <<<<<<<<< Data validation Stage >>>>>>>>>>>:]
[2024-08-15 11:09:56,278: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 11:09:56,279: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 11:09:56,282: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 11:09:56,283: INFO: common: created directory at artifacts:]
[2024-08-15 11:09:56,283: INFO: common: created directory at artifacts/data_validation:]
[2024-08-15 11:09:56,313: INFO: main:  <<<<<<<<< Data validation Stage completed >>>>>>>>>>>:]
[2024-08-15 11:09:56,313: INFO: main: <<<<<<<<<< Data Transformation stage started >>>>>>>>>>>>>:]
[2024-08-15 11:09:56,317: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 11:09:56,318: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 11:09:56,323: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 11:09:56,324: INFO: common: created directory at artifacts:]
[2024-08-15 11:09:56,324: INFO: common: created directory at artifacts/data_transformation:]
[2024-08-15 11:09:56,380: INFO: data_transformation: splitting data into train and test data:]
[2024-08-15 11:09:56,381: INFO: data_transformation: (1199, 12):]
[2024-08-15 11:09:56,381: INFO: data_transformation: (400, 12):]
[2024-08-15 11:09:56,382: INFO: main: <<<<<<<<<< Data Transformation stage completed >>>>>>>>>>>>>:]
[2024-08-15 12:33:05,763: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 12:33:05,769: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:33:05,790: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:33:05,796: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:33:05,852: INFO: common: created directory at artifacts:]
[2024-08-15 12:33:05,853: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 12:33:07,848: INFO: data_ingestion: artifacts/data_ingestion/data.zip download with following headers Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 59B3:1A77D:44DE00:47E6D0:66BE57F3
Accept-Ranges: bytes
Date: Thu, 15 Aug 2024 19:33:07 GMT
Via: 1.1 varnish
X-Served-By: cache-pao-kpao1770025-PAO
X-Cache: MISS
X-Cache-Hits: 0
X-Timer: S1723750388.591034,VS0,VE161
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: a3ef4e167a8c5d4332a3dbfe0a41e07d188e475a
Expires: Thu, 15 Aug 2024 19:38:07 GMT
Source-Age: 0

:]
[2024-08-15 12:33:07,855: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has completed

**********:]
[2024-08-15 12:33:07,856: INFO: main: starting <<<<<<<<< Data validation Stage >>>>>>>>>>>:]
[2024-08-15 12:33:07,858: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:33:07,859: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:33:07,866: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:33:07,867: INFO: common: created directory at artifacts:]
[2024-08-15 12:33:07,868: INFO: common: created directory at artifacts/data_validation:]
[2024-08-15 12:33:07,900: INFO: main:  <<<<<<<<< Data validation Stage completed >>>>>>>>>>>:]
[2024-08-15 12:33:07,900: INFO: main: <<<<<<<<<< Data Transformation stage started >>>>>>>>>>>>>:]
[2024-08-15 12:33:07,903: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:33:07,904: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:33:07,907: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:33:07,907: INFO: common: created directory at artifacts:]
[2024-08-15 12:33:07,908: INFO: common: created directory at artifacts/data_transformation:]
[2024-08-15 12:33:07,971: INFO: data_transformation: splitting data into train and test data:]
[2024-08-15 12:33:07,971: INFO: data_transformation: (1199, 12):]
[2024-08-15 12:33:07,972: INFO: data_transformation: (400, 12):]
[2024-08-15 12:33:07,972: INFO: main: <<<<<<<<<< Data Transformation stage completed >>>>>>>>>>>>>:]
[2024-08-15 12:33:07,972: INFO: main: >>>>>>>> Model training stage started <<<<<<<<:]
[2024-08-15 12:33:07,974: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:33:07,975: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:33:07,991: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:33:07,992: INFO: common: created directory at artifacts:]
[2024-08-15 12:33:07,993: INFO: common: created directory at artifacts/model_trainer:]
[2024-08-15 12:33:07,993: INFO: main: ModelTrainer.__init__() got an unexpected keyword argument 'config':]
[2024-08-15 12:35:38,045: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 12:35:38,052: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:35:38,055: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:35:38,063: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:35:38,064: INFO: common: created directory at artifacts:]
[2024-08-15 12:35:38,065: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 12:35:38,692: INFO: data_ingestion: artifacts/data_ingestion/data.zip download with following headers Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 59B3:1A77D:44DE00:47E6D0:66BE57F3
Accept-Ranges: bytes
Date: Thu, 15 Aug 2024 19:35:38 GMT
Via: 1.1 varnish
X-Served-By: cache-pao-kpao1770057-PAO
X-Cache: HIT
X-Cache-Hits: 0
X-Timer: S1723750539.689030,VS0,VE1
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: f3a912800b7c3aefedf9729e374a69caa2e5bf97
Expires: Thu, 15 Aug 2024 19:40:38 GMT
Source-Age: 151

:]
[2024-08-15 12:35:38,697: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has completed

**********:]
[2024-08-15 12:35:38,697: INFO: main: starting <<<<<<<<< Data validation Stage >>>>>>>>>>>:]
[2024-08-15 12:35:38,702: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:35:38,704: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:35:38,710: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:35:38,711: INFO: common: created directory at artifacts:]
[2024-08-15 12:35:38,712: INFO: common: created directory at artifacts/data_validation:]
[2024-08-15 12:35:38,748: INFO: main:  <<<<<<<<< Data validation Stage completed >>>>>>>>>>>:]
[2024-08-15 12:35:38,748: INFO: main: <<<<<<<<<< Data Transformation stage started >>>>>>>>>>>>>:]
[2024-08-15 12:35:38,759: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:35:38,760: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:35:38,764: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:35:38,777: INFO: common: created directory at artifacts:]
[2024-08-15 12:35:38,779: INFO: common: created directory at artifacts/data_transformation:]
[2024-08-15 12:35:38,897: INFO: data_transformation: splitting data into train and test data:]
[2024-08-15 12:35:38,897: INFO: data_transformation: (1199, 12):]
[2024-08-15 12:35:38,898: INFO: data_transformation: (400, 12):]
[2024-08-15 12:35:38,898: INFO: main: <<<<<<<<<< Data Transformation stage completed >>>>>>>>>>>>>:]
[2024-08-15 12:35:38,900: INFO: main: >>>>>>>> Model training stage started <<<<<<<<:]
[2024-08-15 12:35:38,908: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:35:38,909: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:35:38,912: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:35:38,913: INFO: common: created directory at artifacts:]
[2024-08-15 12:35:38,915: INFO: common: created directory at artifacts/model_trainer:]
[2024-08-15 12:35:38,915: INFO: main: ModelTrainer.__init__() got an unexpected keyword argument 'config':]
[2024-08-15 12:41:52,156: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 12:41:52,161: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:41:52,168: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:41:52,172: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:41:52,176: INFO: common: created directory at artifacts:]
[2024-08-15 12:41:52,177: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 12:41:53,153: INFO: data_ingestion: artifacts/data_ingestion/data.zip download with following headers Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 59B3:1A77D:44DE00:47E6D0:66BE57F3
Accept-Ranges: bytes
Date: Thu, 15 Aug 2024 19:41:52 GMT
Via: 1.1 varnish
X-Served-By: cache-pao-kpao1770077-PAO
X-Cache: HIT
X-Cache-Hits: 0
X-Timer: S1723750913.818851,VS0,VE140
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: 9c6dd1cb253738265c48720404ac74cb516739b9
Expires: Thu, 15 Aug 2024 19:46:52 GMT
Source-Age: 0

:]
[2024-08-15 12:41:53,161: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has completed

**********:]
[2024-08-15 12:41:53,161: INFO: main: starting <<<<<<<<< Data validation Stage >>>>>>>>>>>:]
[2024-08-15 12:41:53,172: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:41:53,175: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:41:53,178: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:41:53,179: INFO: common: created directory at artifacts:]
[2024-08-15 12:41:53,180: INFO: common: created directory at artifacts/data_validation:]
[2024-08-15 12:41:53,283: INFO: main:  <<<<<<<<< Data validation Stage completed >>>>>>>>>>>:]
[2024-08-15 12:41:53,283: INFO: main: <<<<<<<<<< Data Transformation stage started >>>>>>>>>>>>>:]
[2024-08-15 12:41:53,286: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:41:53,288: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:41:53,291: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:41:53,292: INFO: common: created directory at artifacts:]
[2024-08-15 12:41:53,293: INFO: common: created directory at artifacts/data_transformation:]
[2024-08-15 12:41:53,359: INFO: data_transformation: splitting data into train and test data:]
[2024-08-15 12:41:53,359: INFO: data_transformation: (1199, 12):]
[2024-08-15 12:41:53,360: INFO: data_transformation: (400, 12):]
[2024-08-15 12:41:53,360: INFO: main: <<<<<<<<<< Data Transformation stage completed >>>>>>>>>>>>>:]
[2024-08-15 12:41:53,361: INFO: main: >>>>>>>> Model training stage started <<<<<<<<:]
[2024-08-15 12:41:53,368: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:41:53,369: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:41:53,372: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:41:53,373: INFO: common: created directory at artifacts:]
[2024-08-15 12:41:53,374: INFO: common: created directory at artifacts/model_trainer:]
[2024-08-15 12:41:53,374: INFO: main: ModelTrainer.__init__() got an unexpected keyword argument 'config':]
[2024-08-15 12:45:17,003: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 12:45:17,008: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:45:17,015: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:45:17,019: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:45:17,021: INFO: common: created directory at artifacts:]
[2024-08-15 12:45:17,024: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 12:45:17,827: INFO: data_ingestion: artifacts/data_ingestion/data.zip download with following headers Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 59B3:1A77D:44DE00:47E6D0:66BE57F3
Accept-Ranges: bytes
Date: Thu, 15 Aug 2024 19:45:17 GMT
Via: 1.1 varnish
X-Served-By: cache-pao-kpao1770071-PAO
X-Cache: HIT
X-Cache-Hits: 0
X-Timer: S1723751118.853646,VS0,VE1
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: ef26c2ad1165f57e5dd062b7402bf12d579a7649
Expires: Thu, 15 Aug 2024 19:50:17 GMT
Source-Age: 205

:]
[2024-08-15 12:45:17,833: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has completed

**********:]
[2024-08-15 12:45:17,833: INFO: main: starting <<<<<<<<< Data validation Stage >>>>>>>>>>>:]
[2024-08-15 12:45:17,838: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:45:17,840: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:45:17,844: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:45:17,845: INFO: common: created directory at artifacts:]
[2024-08-15 12:45:17,847: INFO: common: created directory at artifacts/data_validation:]
[2024-08-15 12:45:17,882: INFO: main:  <<<<<<<<< Data validation Stage completed >>>>>>>>>>>:]
[2024-08-15 12:45:17,882: INFO: main: <<<<<<<<<< Data Transformation stage started >>>>>>>>>>>>>:]
[2024-08-15 12:45:17,886: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:45:17,887: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:45:17,893: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:45:17,894: INFO: common: created directory at artifacts:]
[2024-08-15 12:45:17,895: INFO: common: created directory at artifacts/data_transformation:]
[2024-08-15 12:45:17,959: INFO: data_transformation: splitting data into train and test data:]
[2024-08-15 12:45:17,959: INFO: data_transformation: (1199, 12):]
[2024-08-15 12:45:17,959: INFO: data_transformation: (400, 12):]
[2024-08-15 12:45:17,959: INFO: main: <<<<<<<<<< Data Transformation stage completed >>>>>>>>>>>>>:]
[2024-08-15 12:45:17,960: INFO: main: >>>>>>>> Model training stage started <<<<<<<<:]
[2024-08-15 12:45:17,962: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:45:17,963: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:45:17,966: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:45:17,967: INFO: common: created directory at artifacts:]
[2024-08-15 12:45:17,967: INFO: common: created directory at artifacts/model_trainer:]
[2024-08-15 12:45:17,967: INFO: main: ModelTrainer.__init__() got an unexpected keyword argument 'config':]
[2024-08-15 12:48:35,470: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 12:48:35,475: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:48:35,481: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:48:35,486: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:48:35,489: INFO: common: created directory at artifacts:]
[2024-08-15 12:48:35,489: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 12:48:36,599: INFO: data_ingestion: artifacts/data_ingestion/data.zip download with following headers Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 59B3:1A77D:44DE00:47E6D0:66BE57F3
Accept-Ranges: bytes
Date: Thu, 15 Aug 2024 19:48:36 GMT
Via: 1.1 varnish
X-Served-By: cache-pao-kpao1770076-PAO
X-Cache: HIT
X-Cache-Hits: 0
X-Timer: S1723751316.330627,VS0,VE159
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: f0ed78be43612cb7069a0504d810bc1527c83529
Expires: Thu, 15 Aug 2024 19:53:36 GMT
Source-Age: 0

:]
[2024-08-15 12:48:36,611: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has completed

**********:]
[2024-08-15 12:48:36,612: INFO: main: starting <<<<<<<<< Data validation Stage >>>>>>>>>>>:]
[2024-08-15 12:48:36,622: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:48:36,625: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:48:36,633: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:48:36,635: INFO: common: created directory at artifacts:]
[2024-08-15 12:48:36,636: INFO: common: created directory at artifacts/data_validation:]
[2024-08-15 12:48:36,677: INFO: main:  <<<<<<<<< Data validation Stage completed >>>>>>>>>>>:]
[2024-08-15 12:48:36,677: INFO: main: <<<<<<<<<< Data Transformation stage started >>>>>>>>>>>>>:]
[2024-08-15 12:48:36,683: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:48:36,685: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:48:36,689: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:48:36,692: INFO: common: created directory at artifacts:]
[2024-08-15 12:48:36,693: INFO: common: created directory at artifacts/data_transformation:]
[2024-08-15 12:48:36,767: INFO: data_transformation: splitting data into train and test data:]
[2024-08-15 12:48:36,767: INFO: data_transformation: (1199, 12):]
[2024-08-15 12:48:36,768: INFO: data_transformation: (400, 12):]
[2024-08-15 12:48:36,768: INFO: main: <<<<<<<<<< Data Transformation stage completed >>>>>>>>>>>>>:]
[2024-08-15 12:48:36,768: INFO: main: >>>>>>>> Model training stage started <<<<<<<<:]
[2024-08-15 12:48:36,771: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:48:36,772: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:48:36,783: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:48:36,784: INFO: common: created directory at artifacts:]
[2024-08-15 12:48:36,784: INFO: common: created directory at artifacts/model_trainer:]
[2024-08-15 12:48:36,831: INFO: main: 'DataFrame' object is not callable:]
[2024-08-15 12:53:07,449: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 12:53:07,454: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:53:07,462: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:53:07,465: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:53:07,470: INFO: common: created directory at artifacts:]
[2024-08-15 12:53:07,471: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 12:53:08,219: INFO: data_ingestion: artifacts/data_ingestion/data.zip download with following headers Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 59B3:1A77D:44DE00:47E6D0:66BE57F3
Accept-Ranges: bytes
Date: Thu, 15 Aug 2024 19:53:08 GMT
Via: 1.1 varnish
X-Served-By: cache-pao-kpao1770037-PAO
X-Cache: HIT
X-Cache-Hits: 0
X-Timer: S1723751588.220877,VS0,VE1
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: ea20436d3b23c24abe9bd890e8ad9f931ac87112
Expires: Thu, 15 Aug 2024 19:58:08 GMT
Source-Age: 272

:]
[2024-08-15 12:53:08,271: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has completed

**********:]
[2024-08-15 12:53:08,272: INFO: main: starting <<<<<<<<< Data validation Stage >>>>>>>>>>>:]
[2024-08-15 12:53:08,276: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:53:08,279: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:53:08,282: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:53:08,284: INFO: common: created directory at artifacts:]
[2024-08-15 12:53:08,285: INFO: common: created directory at artifacts/data_validation:]
[2024-08-15 12:53:08,350: INFO: main:  <<<<<<<<< Data validation Stage completed >>>>>>>>>>>:]
[2024-08-15 12:53:08,351: INFO: main: <<<<<<<<<< Data Transformation stage started >>>>>>>>>>>>>:]
[2024-08-15 12:53:08,354: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:53:08,355: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:53:08,358: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:53:08,359: INFO: common: created directory at artifacts:]
[2024-08-15 12:53:08,359: INFO: common: created directory at artifacts/data_transformation:]
[2024-08-15 12:53:08,477: INFO: data_transformation: splitting data into train and test data:]
[2024-08-15 12:53:08,477: INFO: data_transformation: (1199, 12):]
[2024-08-15 12:53:08,477: INFO: data_transformation: (400, 12):]
[2024-08-15 12:53:08,477: INFO: main: <<<<<<<<<< Data Transformation stage completed >>>>>>>>>>>>>:]
[2024-08-15 12:53:08,478: INFO: main: >>>>>>>> Model training stage started <<<<<<<<:]
[2024-08-15 12:53:08,481: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:53:08,484: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:53:08,487: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:53:08,492: INFO: common: created directory at artifacts:]
[2024-08-15 12:53:08,494: INFO: common: created directory at artifacts/model_trainer:]
[2024-08-15 12:53:08,517: INFO: main: ElasticNet.__init__() got an unexpected keyword argument 'li_ratio':]
[2024-08-15 12:54:24,846: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 12:54:24,850: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:54:24,856: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:54:24,864: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:54:24,866: INFO: common: created directory at artifacts:]
[2024-08-15 12:54:24,866: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 12:54:25,454: INFO: data_ingestion: artifacts/data_ingestion/data.zip download with following headers Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 59B3:1A77D:44DE00:47E6D0:66BE57F3
Accept-Ranges: bytes
Date: Thu, 15 Aug 2024 19:54:25 GMT
Via: 1.1 varnish
X-Served-By: cache-pao-kpao1770040-PAO
X-Cache: HIT
X-Cache-Hits: 0
X-Timer: S1723751665.358479,VS0,VE137
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: 23895143acdc441d84b172d9e5e3be85003b6c85
Expires: Thu, 15 Aug 2024 19:59:25 GMT
Source-Age: 0

:]
[2024-08-15 12:54:25,458: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has completed

**********:]
[2024-08-15 12:54:25,458: INFO: main: starting <<<<<<<<< Data validation Stage >>>>>>>>>>>:]
[2024-08-15 12:54:25,473: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:54:25,476: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:54:25,486: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:54:25,488: INFO: common: created directory at artifacts:]
[2024-08-15 12:54:25,488: INFO: common: created directory at artifacts/data_validation:]
[2024-08-15 12:54:25,527: INFO: main:  <<<<<<<<< Data validation Stage completed >>>>>>>>>>>:]
[2024-08-15 12:54:25,527: INFO: main: <<<<<<<<<< Data Transformation stage started >>>>>>>>>>>>>:]
[2024-08-15 12:54:25,537: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:54:25,539: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:54:25,542: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:54:25,543: INFO: common: created directory at artifacts:]
[2024-08-15 12:54:25,544: INFO: common: created directory at artifacts/data_transformation:]
[2024-08-15 12:54:25,623: INFO: data_transformation: splitting data into train and test data:]
[2024-08-15 12:54:25,624: INFO: data_transformation: (1199, 12):]
[2024-08-15 12:54:25,624: INFO: data_transformation: (400, 12):]
[2024-08-15 12:54:25,625: INFO: main: <<<<<<<<<< Data Transformation stage completed >>>>>>>>>>>>>:]
[2024-08-15 12:54:25,625: INFO: main: >>>>>>>> Model training stage started <<<<<<<<:]
[2024-08-15 12:54:25,630: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:54:25,635: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:54:25,641: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:54:25,643: INFO: common: created directory at artifacts:]
[2024-08-15 12:54:25,644: INFO: common: created directory at artifacts/model_trainer:]
[2024-08-15 12:54:25,686: INFO: main: ElasticNet.__init__() got an unexpected keyword argument 'Li_ratio':]
[2024-08-15 12:57:16,871: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 12:57:16,877: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:57:16,884: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:57:16,887: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:57:16,891: INFO: common: created directory at artifacts:]
[2024-08-15 12:57:16,892: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 12:57:17,521: INFO: data_ingestion: artifacts/data_ingestion/data.zip download with following headers Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 59B3:1A77D:44DE00:47E6D0:66BE57F3
Accept-Ranges: bytes
Date: Thu, 15 Aug 2024 19:57:17 GMT
Via: 1.1 varnish
X-Served-By: cache-pao-kpao1770022-PAO
X-Cache: HIT
X-Cache-Hits: 0
X-Timer: S1723751838.565352,VS0,VE1
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: acc02675890cb4c0fa0c15052220dd51523c629d
Expires: Thu, 15 Aug 2024 20:02:17 GMT
Source-Age: 172

:]
[2024-08-15 12:57:17,525: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has completed

**********:]
[2024-08-15 12:57:17,525: INFO: main: starting <<<<<<<<< Data validation Stage >>>>>>>>>>>:]
[2024-08-15 12:57:17,536: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:57:17,540: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:57:17,545: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:57:17,550: INFO: common: created directory at artifacts:]
[2024-08-15 12:57:17,554: INFO: common: created directory at artifacts/data_validation:]
[2024-08-15 12:57:17,590: INFO: main:  <<<<<<<<< Data validation Stage completed >>>>>>>>>>>:]
[2024-08-15 12:57:17,590: INFO: main: <<<<<<<<<< Data Transformation stage started >>>>>>>>>>>>>:]
[2024-08-15 12:57:17,593: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:57:17,594: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:57:17,601: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:57:17,602: INFO: common: created directory at artifacts:]
[2024-08-15 12:57:17,603: INFO: common: created directory at artifacts/data_transformation:]
[2024-08-15 12:57:17,655: INFO: data_transformation: splitting data into train and test data:]
[2024-08-15 12:57:17,655: INFO: data_transformation: (1199, 12):]
[2024-08-15 12:57:17,655: INFO: data_transformation: (400, 12):]
[2024-08-15 12:57:17,656: INFO: main: <<<<<<<<<< Data Transformation stage completed >>>>>>>>>>>>>:]
[2024-08-15 12:57:17,656: INFO: main: >>>>>>>> Model training stage started <<<<<<<<:]
[2024-08-15 12:57:17,661: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:57:17,665: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:57:17,671: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:57:17,671: INFO: common: created directory at artifacts:]
[2024-08-15 12:57:17,672: INFO: common: created directory at artifacts/model_trainer:]
[2024-08-15 12:57:17,685: INFO: main: ElasticNet.__init__() got an unexpected keyword argument 'randon_seed':]
[2024-08-15 12:58:03,807: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 12:58:03,811: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:58:03,820: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:58:03,827: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:58:03,829: INFO: common: created directory at artifacts:]
[2024-08-15 12:58:03,832: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 12:58:04,320: INFO: data_ingestion: artifacts/data_ingestion/data.zip download with following headers Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 59B3:1A77D:44DE00:47E6D0:66BE57F3
Accept-Ranges: bytes
Date: Thu, 15 Aug 2024 19:58:04 GMT
Via: 1.1 varnish
X-Served-By: cache-pao-kpao1770065-PAO
X-Cache: HIT
X-Cache-Hits: 1
X-Timer: S1723751884.305170,VS0,VE1
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: 5736a460686d516c75b2a35f2bd586922acf1424
Expires: Thu, 15 Aug 2024 20:03:04 GMT
Source-Age: 219

:]
[2024-08-15 12:58:04,325: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has completed

**********:]
[2024-08-15 12:58:04,325: INFO: main: starting <<<<<<<<< Data validation Stage >>>>>>>>>>>:]
[2024-08-15 12:58:04,341: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:58:04,343: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:58:04,354: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:58:04,358: INFO: common: created directory at artifacts:]
[2024-08-15 12:58:04,359: INFO: common: created directory at artifacts/data_validation:]
[2024-08-15 12:58:04,483: INFO: main:  <<<<<<<<< Data validation Stage completed >>>>>>>>>>>:]
[2024-08-15 12:58:04,484: INFO: main: <<<<<<<<<< Data Transformation stage started >>>>>>>>>>>>>:]
[2024-08-15 12:58:04,492: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:58:04,494: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:58:04,498: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:58:04,499: INFO: common: created directory at artifacts:]
[2024-08-15 12:58:04,499: INFO: common: created directory at artifacts/data_transformation:]
[2024-08-15 12:58:04,601: INFO: data_transformation: splitting data into train and test data:]
[2024-08-15 12:58:04,601: INFO: data_transformation: (1199, 12):]
[2024-08-15 12:58:04,601: INFO: data_transformation: (400, 12):]
[2024-08-15 12:58:04,602: INFO: main: <<<<<<<<<< Data Transformation stage completed >>>>>>>>>>>>>:]
[2024-08-15 12:58:04,602: INFO: main: >>>>>>>> Model training stage started <<<<<<<<:]
[2024-08-15 12:58:04,606: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 12:58:04,613: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 12:58:04,621: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 12:58:04,625: INFO: common: created directory at artifacts:]
[2024-08-15 12:58:04,627: INFO: common: created directory at artifacts/model_trainer:]
[2024-08-15 12:58:04,680: INFO: main: >>>>>>>> Model training stage completed <<<<<<<<:]
[2024-08-15 20:26:54,844: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 20:26:54,859: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:26:54,869: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:26:54,875: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:26:54,883: INFO: common: created directory at artifacts:]
[2024-08-15 20:26:54,884: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 20:26:55,994: INFO: data_ingestion: artifacts/data_ingestion/data.zip download with following headers Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: F880:70F30:7ACEF6:801944:66BEC6FF
Accept-Ranges: bytes
Date: Fri, 16 Aug 2024 03:26:56 GMT
Via: 1.1 varnish
X-Served-By: cache-pao-kpao1770033-PAO
X-Cache: MISS
X-Cache-Hits: 0
X-Timer: S1723778817.853758,VS0,VE146
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: 2f3c00183a1fc2964a63b7358c3fd666934e0088
Expires: Fri, 16 Aug 2024 03:31:56 GMT
Source-Age: 0

:]
[2024-08-15 20:26:56,002: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has completed

**********:]
[2024-08-15 20:26:56,003: INFO: main: starting <<<<<<<<< Data validation Stage >>>>>>>>>>>:]
[2024-08-15 20:26:56,021: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:26:56,023: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:26:56,035: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:26:56,036: INFO: common: created directory at artifacts:]
[2024-08-15 20:26:56,036: INFO: common: created directory at artifacts/data_validation:]
[2024-08-15 20:26:56,104: INFO: main:  <<<<<<<<< Data validation Stage completed >>>>>>>>>>>:]
[2024-08-15 20:26:56,107: INFO: main: <<<<<<<<<< Data Transformation stage started >>>>>>>>>>>>>:]
[2024-08-15 20:26:56,117: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:26:56,118: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:26:56,120: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:26:56,121: INFO: common: created directory at artifacts:]
[2024-08-15 20:26:56,121: INFO: common: created directory at artifacts/data_transformation:]
[2024-08-15 20:26:56,187: INFO: data_transformation: splitting data into train and test data:]
[2024-08-15 20:26:56,188: INFO: data_transformation: (1199, 12):]
[2024-08-15 20:26:56,188: INFO: data_transformation: (400, 12):]
[2024-08-15 20:26:56,191: INFO: main: <<<<<<<<<< Data Transformation stage completed >>>>>>>>>>>>>:]
[2024-08-15 20:26:56,191: INFO: main: >>>>>>>> Model training stage started <<<<<<<<:]
[2024-08-15 20:26:56,196: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:26:56,197: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:26:56,199: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:26:56,199: INFO: common: created directory at artifacts:]
[2024-08-15 20:26:56,200: INFO: common: created directory at artifacts/model_trainer:]
[2024-08-15 20:26:56,247: INFO: main: >>>>>>>> Model training stage completed <<<<<<<<:]
[2024-08-15 20:26:56,247: INFO: main: >>>>>>>>>>Model Evaluation stage started <<<<<<<<<<:]
[2024-08-15 20:26:56,251: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:26:56,252: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:26:56,268: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:26:56,269: INFO: common: created directory at artifacts:]
[2024-08-15 20:26:56,269: ERROR: main: cannot access local variable 'schema' where it is not associated with a value:]
Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/main.py", line 60, in <module>
    obj.main()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/pipeline/stage_05_model_evaluation.py", line 16, in main
    model_evaluation_config = config.get_model_evaluation_config()
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/config/configuration.py", line 84, in get_model_evaluation_config
    schema = schema.TARGET_COLUMN
             ^^^^^^
UnboundLocalError: cannot access local variable 'schema' where it is not associated with a value
[2024-08-15 20:27:30,028: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 20:27:30,037: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:27:30,053: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:27:30,089: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:27:30,101: INFO: common: created directory at artifacts:]
[2024-08-15 20:27:30,102: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 20:27:30,576: INFO: data_ingestion: artifacts/data_ingestion/data.zip download with following headers Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: F880:70F30:7ACEF6:801944:66BEC6FF
Accept-Ranges: bytes
Date: Fri, 16 Aug 2024 03:27:31 GMT
Via: 1.1 varnish
X-Served-By: cache-pao-kpao1770082-PAO
X-Cache: HIT
X-Cache-Hits: 1
X-Timer: S1723778852.695617,VS0,VE1
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: 77993d3e1316d5b2dec98fab2f33f3a5ca41752e
Expires: Fri, 16 Aug 2024 03:32:31 GMT
Source-Age: 35

:]
[2024-08-15 20:27:30,580: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has completed

**********:]
[2024-08-15 20:27:30,580: INFO: main: starting <<<<<<<<< Data validation Stage >>>>>>>>>>>:]
[2024-08-15 20:27:30,588: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:27:30,607: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:27:30,610: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:27:30,612: INFO: common: created directory at artifacts:]
[2024-08-15 20:27:30,613: INFO: common: created directory at artifacts/data_validation:]
[2024-08-15 20:27:30,652: INFO: main:  <<<<<<<<< Data validation Stage completed >>>>>>>>>>>:]
[2024-08-15 20:27:30,652: INFO: main: <<<<<<<<<< Data Transformation stage started >>>>>>>>>>>>>:]
[2024-08-15 20:27:30,664: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:27:30,667: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:27:30,669: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:27:30,670: INFO: common: created directory at artifacts:]
[2024-08-15 20:27:30,670: INFO: common: created directory at artifacts/data_transformation:]
[2024-08-15 20:27:30,786: INFO: data_transformation: splitting data into train and test data:]
[2024-08-15 20:27:30,787: INFO: data_transformation: (1199, 12):]
[2024-08-15 20:27:30,787: INFO: data_transformation: (400, 12):]
[2024-08-15 20:27:30,787: INFO: main: <<<<<<<<<< Data Transformation stage completed >>>>>>>>>>>>>:]
[2024-08-15 20:27:30,787: INFO: main: >>>>>>>> Model training stage started <<<<<<<<:]
[2024-08-15 20:27:30,799: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:27:30,800: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:27:30,802: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:27:30,804: INFO: common: created directory at artifacts:]
[2024-08-15 20:27:30,805: INFO: common: created directory at artifacts/model_trainer:]
[2024-08-15 20:27:30,853: INFO: main: >>>>>>>> Model training stage completed <<<<<<<<:]
[2024-08-15 20:27:30,854: INFO: main: >>>>>>>>>>Model Evaluation stage started <<<<<<<<<<:]
[2024-08-15 20:27:30,864: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:27:30,866: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:27:30,868: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:27:30,869: INFO: common: created directory at artifacts:]
[2024-08-15 20:27:30,871: ERROR: main: ModelEvaluationConfig.__init__() got an unexpected keyword argument 'model_path':]
Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/main.py", line 60, in <module>
    obj.main()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/pipeline/stage_05_model_evaluation.py", line 16, in main
    model_evaluation_config = config.get_model_evaluation_config()
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/config/configuration.py", line 86, in get_model_evaluation_config
    model_evaluation_config = ModelEvaluationConfig(
                              ^^^^^^^^^^^^^^^^^^^^^^
TypeError: ModelEvaluationConfig.__init__() got an unexpected keyword argument 'model_path'
[2024-08-15 20:28:06,058: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 20:28:06,065: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:28:06,068: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:28:06,071: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:28:06,074: INFO: common: created directory at artifacts:]
[2024-08-15 20:28:06,075: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 20:28:06,651: INFO: data_ingestion: artifacts/data_ingestion/data.zip download with following headers Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: F880:70F30:7ACEF6:801944:66BEC6FF
Accept-Ranges: bytes
Date: Fri, 16 Aug 2024 03:28:07 GMT
Via: 1.1 varnish
X-Served-By: cache-pao-kpao1770050-PAO
X-Cache: HIT
X-Cache-Hits: 1
X-Timer: S1723778888.671915,VS0,VE2
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: c7e0e41e0df1d03c74f7d7425270eceb08bb6bc6
Expires: Fri, 16 Aug 2024 03:33:07 GMT
Source-Age: 71

:]
[2024-08-15 20:28:06,657: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has completed

**********:]
[2024-08-15 20:28:06,658: INFO: main: starting <<<<<<<<< Data validation Stage >>>>>>>>>>>:]
[2024-08-15 20:28:06,669: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:28:06,670: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:28:06,672: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:28:06,673: INFO: common: created directory at artifacts:]
[2024-08-15 20:28:06,681: INFO: common: created directory at artifacts/data_validation:]
[2024-08-15 20:28:06,735: INFO: main:  <<<<<<<<< Data validation Stage completed >>>>>>>>>>>:]
[2024-08-15 20:28:06,735: INFO: main: <<<<<<<<<< Data Transformation stage started >>>>>>>>>>>>>:]
[2024-08-15 20:28:06,740: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:28:06,744: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:28:06,746: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:28:06,747: INFO: common: created directory at artifacts:]
[2024-08-15 20:28:06,748: INFO: common: created directory at artifacts/data_transformation:]
[2024-08-15 20:28:06,826: INFO: data_transformation: splitting data into train and test data:]
[2024-08-15 20:28:06,827: INFO: data_transformation: (1199, 12):]
[2024-08-15 20:28:06,827: INFO: data_transformation: (400, 12):]
[2024-08-15 20:28:06,828: INFO: main: <<<<<<<<<< Data Transformation stage completed >>>>>>>>>>>>>:]
[2024-08-15 20:28:06,828: INFO: main: >>>>>>>> Model training stage started <<<<<<<<:]
[2024-08-15 20:28:06,832: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:28:06,834: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:28:06,836: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:28:06,836: INFO: common: created directory at artifacts:]
[2024-08-15 20:28:06,837: INFO: common: created directory at artifacts/model_trainer:]
[2024-08-15 20:28:06,878: INFO: main: >>>>>>>> Model training stage completed <<<<<<<<:]
[2024-08-15 20:28:06,879: INFO: main: >>>>>>>>>>Model Evaluation stage started <<<<<<<<<<:]
[2024-08-15 20:28:06,885: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:28:06,887: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:28:06,889: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:28:06,889: INFO: common: created directory at artifacts:]
[2024-08-15 20:28:06,895: ERROR: main: ModelEvaluationConfig.__init__() got an unexpected keyword argument 'model_path':]
Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/main.py", line 60, in <module>
    obj.main()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/pipeline/stage_05_model_evaluation.py", line 16, in main
    model_evaluation_config = config.get_model_evaluation_config()
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/config/configuration.py", line 86, in get_model_evaluation_config
    model_evaluation_config = ModelEvaluationConfig(
                              ^^^^^^^^^^^^^^^^^^^^^^
TypeError: ModelEvaluationConfig.__init__() got an unexpected keyword argument 'model_path'
[2024-08-15 20:28:44,615: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 20:28:44,620: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:28:44,623: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:28:44,626: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:28:44,631: INFO: common: created directory at artifacts:]
[2024-08-15 20:28:44,633: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 20:28:45,367: INFO: data_ingestion: artifacts/data_ingestion/data.zip download with following headers Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: F880:70F30:7ACEF6:801944:66BEC6FF
Accept-Ranges: bytes
Date: Fri, 16 Aug 2024 03:28:46 GMT
Via: 1.1 varnish
X-Served-By: cache-pao-kpao1770057-PAO
X-Cache: HIT
X-Cache-Hits: 1
X-Timer: S1723778926.453033,VS0,VE1
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: f42785d2eaa70352b75800ee9f80845fe01d97c9
Expires: Fri, 16 Aug 2024 03:33:46 GMT
Source-Age: 109

:]
[2024-08-15 20:28:45,371: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has completed

**********:]
[2024-08-15 20:28:45,371: INFO: main: starting <<<<<<<<< Data validation Stage >>>>>>>>>>>:]
[2024-08-15 20:28:45,384: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:28:45,386: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:28:45,388: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:28:45,389: INFO: common: created directory at artifacts:]
[2024-08-15 20:28:45,398: INFO: common: created directory at artifacts/data_validation:]
[2024-08-15 20:28:45,436: INFO: main:  <<<<<<<<< Data validation Stage completed >>>>>>>>>>>:]
[2024-08-15 20:28:45,436: INFO: main: <<<<<<<<<< Data Transformation stage started >>>>>>>>>>>>>:]
[2024-08-15 20:28:45,448: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:28:45,450: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:28:45,452: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:28:45,453: INFO: common: created directory at artifacts:]
[2024-08-15 20:28:45,453: INFO: common: created directory at artifacts/data_transformation:]
[2024-08-15 20:28:45,515: INFO: data_transformation: splitting data into train and test data:]
[2024-08-15 20:28:45,515: INFO: data_transformation: (1199, 12):]
[2024-08-15 20:28:45,516: INFO: data_transformation: (400, 12):]
[2024-08-15 20:28:45,516: INFO: main: <<<<<<<<<< Data Transformation stage completed >>>>>>>>>>>>>:]
[2024-08-15 20:28:45,516: INFO: main: >>>>>>>> Model training stage started <<<<<<<<:]
[2024-08-15 20:28:45,519: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:28:45,521: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:28:45,524: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:28:45,526: INFO: common: created directory at artifacts:]
[2024-08-15 20:28:45,527: INFO: common: created directory at artifacts/model_trainer:]
[2024-08-15 20:28:45,567: INFO: main: >>>>>>>> Model training stage completed <<<<<<<<:]
[2024-08-15 20:28:45,568: INFO: main: >>>>>>>>>>Model Evaluation stage started <<<<<<<<<<:]
[2024-08-15 20:28:45,572: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:28:45,578: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:28:45,581: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:28:45,582: INFO: common: created directory at artifacts:]
[2024-08-15 20:28:45,582: ERROR: main: ModelEvaluationConfig.__init__() got an unexpected keyword argument 'model_path':]
Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/main.py", line 60, in <module>
    obj.main()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/pipeline/stage_05_model_evaluation.py", line 16, in main
    model_evaluation_config = config.get_model_evaluation_config()
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/config/configuration.py", line 86, in get_model_evaluation_config
    model_evaluation_config = ModelEvaluationConfig(
                              ^^^^^^^^^^^^^^^^^^^^^^
TypeError: ModelEvaluationConfig.__init__() got an unexpected keyword argument 'model_path'
[2024-08-15 20:29:39,105: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 20:29:39,110: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:29:39,119: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:29:39,135: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:29:39,138: INFO: common: created directory at artifacts:]
[2024-08-15 20:29:39,145: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 20:29:39,726: INFO: data_ingestion: artifacts/data_ingestion/data.zip download with following headers Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: F880:70F30:7ACEF6:801944:66BEC6FF
Accept-Ranges: bytes
Date: Fri, 16 Aug 2024 03:29:40 GMT
Via: 1.1 varnish
X-Served-By: cache-pao-kpao1770060-PAO
X-Cache: HIT
X-Cache-Hits: 1
X-Timer: S1723778981.830052,VS0,VE1
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: 0679a6633bb1d20f239c9d5ca28a7d1fdbdee6b0
Expires: Fri, 16 Aug 2024 03:34:40 GMT
Source-Age: 164

:]
[2024-08-15 20:29:39,736: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has completed

**********:]
[2024-08-15 20:29:39,738: INFO: main: starting <<<<<<<<< Data validation Stage >>>>>>>>>>>:]
[2024-08-15 20:29:39,752: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:29:39,763: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:29:39,781: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:29:39,782: INFO: common: created directory at artifacts:]
[2024-08-15 20:29:39,783: INFO: common: created directory at artifacts/data_validation:]
[2024-08-15 20:29:39,886: INFO: main:  <<<<<<<<< Data validation Stage completed >>>>>>>>>>>:]
[2024-08-15 20:29:39,886: INFO: main: <<<<<<<<<< Data Transformation stage started >>>>>>>>>>>>>:]
[2024-08-15 20:29:39,898: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:29:39,900: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:29:39,903: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:29:39,904: INFO: common: created directory at artifacts:]
[2024-08-15 20:29:39,904: INFO: common: created directory at artifacts/data_transformation:]
[2024-08-15 20:29:39,988: INFO: data_transformation: splitting data into train and test data:]
[2024-08-15 20:29:39,989: INFO: data_transformation: (1199, 12):]
[2024-08-15 20:29:39,991: INFO: data_transformation: (400, 12):]
[2024-08-15 20:29:39,996: INFO: main: <<<<<<<<<< Data Transformation stage completed >>>>>>>>>>>>>:]
[2024-08-15 20:29:39,997: INFO: main: >>>>>>>> Model training stage started <<<<<<<<:]
[2024-08-15 20:29:40,022: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:29:40,031: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:29:40,034: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:29:40,035: INFO: common: created directory at artifacts:]
[2024-08-15 20:29:40,036: INFO: common: created directory at artifacts/model_trainer:]
[2024-08-15 20:29:40,091: INFO: main: >>>>>>>> Model training stage completed <<<<<<<<:]
[2024-08-15 20:29:40,091: INFO: main: >>>>>>>>>>Model Evaluation stage started <<<<<<<<<<:]
[2024-08-15 20:29:40,099: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:29:40,102: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:29:40,113: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:29:40,114: INFO: common: created directory at artifacts:]
[2024-08-15 20:29:40,115: ERROR: main: 'ModelEvaluationConfig' object has no attribute 'test_dath_path':]
Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/main.py", line 60, in <module>
    obj.main()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/pipeline/stage_05_model_evaluation.py", line 18, in main
    model_evaluation.log_into_mlflow()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/components/model_evaluation.py", line 26, in log_into_mlflow
    test_data = pd.read_csv(self.config.test_dath_path)
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'ModelEvaluationConfig' object has no attribute 'test_dath_path'. Did you mean: 'test_data_path'?
[2024-08-15 20:30:35,218: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 20:30:35,223: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:30:35,226: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:30:35,230: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:30:35,231: INFO: common: created directory at artifacts:]
[2024-08-15 20:30:35,234: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 20:30:35,792: INFO: data_ingestion: artifacts/data_ingestion/data.zip download with following headers Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: F880:70F30:7ACEF6:801944:66BEC6FF
Accept-Ranges: bytes
Date: Fri, 16 Aug 2024 03:30:36 GMT
Via: 1.1 varnish
X-Served-By: cache-pao-kpao1770042-PAO
X-Cache: HIT
X-Cache-Hits: 1
X-Timer: S1723779037.918147,VS0,VE1
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: a85a23eec48dbf38b7b2222d64d8da05f310aa7f
Expires: Fri, 16 Aug 2024 03:35:36 GMT
Source-Age: 220

:]
[2024-08-15 20:30:35,797: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has completed

**********:]
[2024-08-15 20:30:35,797: INFO: main: starting <<<<<<<<< Data validation Stage >>>>>>>>>>>:]
[2024-08-15 20:30:35,804: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:30:35,816: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:30:35,819: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:30:35,819: INFO: common: created directory at artifacts:]
[2024-08-15 20:30:35,820: INFO: common: created directory at artifacts/data_validation:]
[2024-08-15 20:30:35,879: INFO: main:  <<<<<<<<< Data validation Stage completed >>>>>>>>>>>:]
[2024-08-15 20:30:35,880: INFO: main: <<<<<<<<<< Data Transformation stage started >>>>>>>>>>>>>:]
[2024-08-15 20:30:35,884: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:30:35,886: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:30:35,889: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:30:35,890: INFO: common: created directory at artifacts:]
[2024-08-15 20:30:35,891: INFO: common: created directory at artifacts/data_transformation:]
[2024-08-15 20:30:36,064: INFO: data_transformation: splitting data into train and test data:]
[2024-08-15 20:30:36,064: INFO: data_transformation: (1199, 12):]
[2024-08-15 20:30:36,065: INFO: data_transformation: (400, 12):]
[2024-08-15 20:30:36,065: INFO: main: <<<<<<<<<< Data Transformation stage completed >>>>>>>>>>>>>:]
[2024-08-15 20:30:36,066: INFO: main: >>>>>>>> Model training stage started <<<<<<<<:]
[2024-08-15 20:30:36,076: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:30:36,083: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:30:36,088: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:30:36,089: INFO: common: created directory at artifacts:]
[2024-08-15 20:30:36,092: INFO: common: created directory at artifacts/model_trainer:]
[2024-08-15 20:30:36,131: INFO: main: >>>>>>>> Model training stage completed <<<<<<<<:]
[2024-08-15 20:30:36,131: INFO: main: >>>>>>>>>>Model Evaluation stage started <<<<<<<<<<:]
[2024-08-15 20:30:36,135: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:30:36,138: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:30:36,149: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:30:36,150: INFO: common: created directory at artifacts:]
[2024-08-15 20:30:36,169: ERROR: main: module 'mlflow' has no attribute 'set_registry_url':]
Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/main.py", line 60, in <module>
    obj.main()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/pipeline/stage_05_model_evaluation.py", line 18, in main
    model_evaluation.log_into_mlflow()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/components/model_evaluation.py", line 32, in log_into_mlflow
    mlflow.set_registry_url(self.config.mlflow_url)
    ^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: module 'mlflow' has no attribute 'set_registry_url'. Did you mean: 'set_registry_uri'?
[2024-08-15 20:31:06,922: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 20:31:06,929: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:31:06,932: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:31:06,935: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:31:06,936: INFO: common: created directory at artifacts:]
[2024-08-15 20:31:06,936: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 20:31:07,403: INFO: data_ingestion: artifacts/data_ingestion/data.zip download with following headers Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: F880:70F30:7ACEF6:801944:66BEC6FF
Accept-Ranges: bytes
Date: Fri, 16 Aug 2024 03:31:08 GMT
Via: 1.1 varnish
X-Served-By: cache-pao-kpao1770042-PAO
X-Cache: HIT
X-Cache-Hits: 2
X-Timer: S1723779069.527931,VS0,VE0
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: 4539ee61f12d02f7d552f44a21ba82fae553682b
Expires: Fri, 16 Aug 2024 03:36:08 GMT
Source-Age: 252

:]
[2024-08-15 20:31:07,407: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has completed

**********:]
[2024-08-15 20:31:07,407: INFO: main: starting <<<<<<<<< Data validation Stage >>>>>>>>>>>:]
[2024-08-15 20:31:07,413: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:31:07,418: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:31:07,421: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:31:07,422: INFO: common: created directory at artifacts:]
[2024-08-15 20:31:07,422: INFO: common: created directory at artifacts/data_validation:]
[2024-08-15 20:31:07,465: INFO: main:  <<<<<<<<< Data validation Stage completed >>>>>>>>>>>:]
[2024-08-15 20:31:07,466: INFO: main: <<<<<<<<<< Data Transformation stage started >>>>>>>>>>>>>:]
[2024-08-15 20:31:07,469: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:31:07,470: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:31:07,473: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:31:07,474: INFO: common: created directory at artifacts:]
[2024-08-15 20:31:07,475: INFO: common: created directory at artifacts/data_transformation:]
[2024-08-15 20:31:07,549: INFO: data_transformation: splitting data into train and test data:]
[2024-08-15 20:31:07,550: INFO: data_transformation: (1199, 12):]
[2024-08-15 20:31:07,552: INFO: data_transformation: (400, 12):]
[2024-08-15 20:31:07,553: INFO: main: <<<<<<<<<< Data Transformation stage completed >>>>>>>>>>>>>:]
[2024-08-15 20:31:07,554: INFO: main: >>>>>>>> Model training stage started <<<<<<<<:]
[2024-08-15 20:31:07,569: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:31:07,571: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:31:07,578: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:31:07,579: INFO: common: created directory at artifacts:]
[2024-08-15 20:31:07,580: INFO: common: created directory at artifacts/model_trainer:]
[2024-08-15 20:31:07,622: INFO: main: >>>>>>>> Model training stage completed <<<<<<<<:]
[2024-08-15 20:31:07,623: INFO: main: >>>>>>>>>>Model Evaluation stage started <<<<<<<<<<:]
[2024-08-15 20:31:07,634: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:31:07,635: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:31:07,637: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:31:07,638: INFO: common: created directory at artifacts:]
[2024-08-15 20:31:07,654: ERROR: main: module 'mlflow' has no attribute 'get_tracking_url':]
Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/main.py", line 60, in <module>
    obj.main()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/pipeline/stage_05_model_evaluation.py", line 18, in main
    model_evaluation.log_into_mlflow()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/components/model_evaluation.py", line 33, in log_into_mlflow
    tracking_url_type_store = urlparse(mlflow.get_tracking_url()).schema
                                       ^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: module 'mlflow' has no attribute 'get_tracking_url'. Did you mean: 'get_tracking_uri'?
[2024-08-15 20:32:00,151: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 20:32:00,157: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:32:00,163: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:32:00,169: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:32:00,172: INFO: common: created directory at artifacts:]
[2024-08-15 20:32:00,173: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 20:32:00,816: INFO: data_ingestion: artifacts/data_ingestion/data.zip download with following headers Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: F880:70F30:7ACEF6:801944:66BEC6FF
Accept-Ranges: bytes
Date: Fri, 16 Aug 2024 03:32:01 GMT
Via: 1.1 varnish
X-Served-By: cache-pao-kpao1770054-PAO
X-Cache: HIT
X-Cache-Hits: 0
X-Timer: S1723779122.805883,VS0,VE139
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: bd27a3381d01192df993aa14f6409c69cfd3c584
Expires: Fri, 16 Aug 2024 03:37:01 GMT
Source-Age: 0

:]
[2024-08-15 20:32:00,833: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has completed

**********:]
[2024-08-15 20:32:00,834: INFO: main: starting <<<<<<<<< Data validation Stage >>>>>>>>>>>:]
[2024-08-15 20:32:00,837: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:32:00,838: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:32:00,841: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:32:00,849: INFO: common: created directory at artifacts:]
[2024-08-15 20:32:00,850: INFO: common: created directory at artifacts/data_validation:]
[2024-08-15 20:32:00,913: INFO: main:  <<<<<<<<< Data validation Stage completed >>>>>>>>>>>:]
[2024-08-15 20:32:00,915: INFO: main: <<<<<<<<<< Data Transformation stage started >>>>>>>>>>>>>:]
[2024-08-15 20:32:00,919: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:32:00,921: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:32:00,923: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:32:00,924: INFO: common: created directory at artifacts:]
[2024-08-15 20:32:00,927: INFO: common: created directory at artifacts/data_transformation:]
[2024-08-15 20:32:01,006: INFO: data_transformation: splitting data into train and test data:]
[2024-08-15 20:32:01,006: INFO: data_transformation: (1199, 12):]
[2024-08-15 20:32:01,007: INFO: data_transformation: (400, 12):]
[2024-08-15 20:32:01,008: INFO: main: <<<<<<<<<< Data Transformation stage completed >>>>>>>>>>>>>:]
[2024-08-15 20:32:01,008: INFO: main: >>>>>>>> Model training stage started <<<<<<<<:]
[2024-08-15 20:32:01,011: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:32:01,014: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:32:01,016: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:32:01,017: INFO: common: created directory at artifacts:]
[2024-08-15 20:32:01,017: INFO: common: created directory at artifacts/model_trainer:]
[2024-08-15 20:32:01,055: INFO: main: >>>>>>>> Model training stage completed <<<<<<<<:]
[2024-08-15 20:32:01,055: INFO: main: >>>>>>>>>>Model Evaluation stage started <<<<<<<<<<:]
[2024-08-15 20:32:01,061: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:32:01,064: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:32:01,067: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:32:01,068: INFO: common: created directory at artifacts:]
[2024-08-15 20:32:01,085: ERROR: main: 'ParseResult' object has no attribute 'schema':]
Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/main.py", line 60, in <module>
    obj.main()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/pipeline/stage_05_model_evaluation.py", line 18, in main
    model_evaluation.log_into_mlflow()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/components/model_evaluation.py", line 33, in log_into_mlflow
    tracking_url_type_store = urlparse(mlflow.get_tracking_uri()).schema
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'ParseResult' object has no attribute 'schema'. Did you mean: 'scheme'?
[2024-08-15 20:33:25,857: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 20:33:25,864: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:33:25,873: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:33:25,878: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:33:25,881: INFO: common: created directory at artifacts:]
[2024-08-15 20:33:25,881: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 20:33:26,322: INFO: data_ingestion: artifacts/data_ingestion/data.zip download with following headers Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: F880:70F30:7ACEF6:801944:66BEC6FF
Accept-Ranges: bytes
Date: Fri, 16 Aug 2024 03:33:27 GMT
Via: 1.1 varnish
X-Served-By: cache-pao-kpao1770070-PAO
X-Cache: HIT
X-Cache-Hits: 0
X-Timer: S1723779207.451078,VS0,VE1
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: 227c15acf6965eea259eb39008b3b4779f05dbc6
Expires: Fri, 16 Aug 2024 03:38:27 GMT
Source-Age: 86

:]
[2024-08-15 20:33:26,328: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has completed

**********:]
[2024-08-15 20:33:26,329: INFO: main: starting <<<<<<<<< Data validation Stage >>>>>>>>>>>:]
[2024-08-15 20:33:26,337: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:33:26,342: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:33:26,349: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:33:26,351: INFO: common: created directory at artifacts:]
[2024-08-15 20:33:26,352: INFO: common: created directory at artifacts/data_validation:]
[2024-08-15 20:33:26,382: INFO: main:  <<<<<<<<< Data validation Stage completed >>>>>>>>>>>:]
[2024-08-15 20:33:26,382: INFO: main: <<<<<<<<<< Data Transformation stage started >>>>>>>>>>>>>:]
[2024-08-15 20:33:26,385: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:33:26,386: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:33:26,390: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:33:26,391: INFO: common: created directory at artifacts:]
[2024-08-15 20:33:26,394: INFO: common: created directory at artifacts/data_transformation:]
[2024-08-15 20:33:26,449: INFO: data_transformation: splitting data into train and test data:]
[2024-08-15 20:33:26,450: INFO: data_transformation: (1199, 12):]
[2024-08-15 20:33:26,450: INFO: data_transformation: (400, 12):]
[2024-08-15 20:33:26,450: INFO: main: <<<<<<<<<< Data Transformation stage completed >>>>>>>>>>>>>:]
[2024-08-15 20:33:26,450: INFO: main: >>>>>>>> Model training stage started <<<<<<<<:]
[2024-08-15 20:33:26,454: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:33:26,455: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:33:26,460: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:33:26,461: INFO: common: created directory at artifacts:]
[2024-08-15 20:33:26,461: INFO: common: created directory at artifacts/model_trainer:]
[2024-08-15 20:33:26,521: INFO: main: >>>>>>>> Model training stage completed <<<<<<<<:]
[2024-08-15 20:33:26,521: INFO: main: >>>>>>>>>>Model Evaluation stage started <<<<<<<<<<:]
[2024-08-15 20:33:26,537: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:33:26,539: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:33:26,541: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:33:26,544: INFO: common: created directory at artifacts:]
[2024-08-15 20:33:28,486: ERROR: main: [Errno 2] No such file or directory: 'artifacts/model_evaluation/metrics.json':]
Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/main.py", line 60, in <module>
    obj.main()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/pipeline/stage_05_model_evaluation.py", line 18, in main
    model_evaluation.log_into_mlflow()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/components/model_evaluation.py", line 40, in log_into_mlflow
    save_json(path=Path(self.config.metric_file_name), data=scores)
  File "/Users/deepa/.local/pip3/venv/lib/python3.12/site-packages/ensure/main.py", line 835, in __call__
    return self.f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/utils/common.py", line 44, in save_json
    with open(path, "w") as f:
         ^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: 'artifacts/model_evaluation/metrics.json'
[2024-08-15 20:38:48,727: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 20:38:48,733: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:38:48,740: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:38:48,750: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:38:48,756: INFO: common: created directory at artifacts:]
[2024-08-15 20:38:48,757: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 20:38:51,358: INFO: data_ingestion: artifacts/data_ingestion/data.zip download with following headers Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: F880:70F30:7ACEF6:801944:66BEC6FF
Accept-Ranges: bytes
Date: Fri, 16 Aug 2024 03:38:52 GMT
Via: 1.1 varnish
X-Served-By: cache-pao-kpao1770062-PAO
X-Cache: HIT
X-Cache-Hits: 0
X-Timer: S1723779532.208557,VS0,VE121
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: 57a357ed795daed279cf93f0f7a3332870234895
Expires: Fri, 16 Aug 2024 03:43:52 GMT
Source-Age: 0

:]
[2024-08-15 20:38:51,379: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has completed

**********:]
[2024-08-15 20:38:51,380: INFO: main: starting <<<<<<<<< Data validation Stage >>>>>>>>>>>:]
[2024-08-15 20:38:51,386: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:38:51,388: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:38:51,390: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:38:51,391: INFO: common: created directory at artifacts:]
[2024-08-15 20:38:51,391: INFO: common: created directory at artifacts/data_validation:]
[2024-08-15 20:38:51,435: INFO: main:  <<<<<<<<< Data validation Stage completed >>>>>>>>>>>:]
[2024-08-15 20:38:51,436: INFO: main: <<<<<<<<<< Data Transformation stage started >>>>>>>>>>>>>:]
[2024-08-15 20:38:51,439: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:38:51,440: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:38:51,443: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:38:51,443: INFO: common: created directory at artifacts:]
[2024-08-15 20:38:51,444: INFO: common: created directory at artifacts/data_transformation:]
[2024-08-15 20:38:51,530: INFO: data_transformation: splitting data into train and test data:]
[2024-08-15 20:38:51,530: INFO: data_transformation: (1199, 12):]
[2024-08-15 20:38:51,530: INFO: data_transformation: (400, 12):]
[2024-08-15 20:38:51,531: INFO: main: <<<<<<<<<< Data Transformation stage completed >>>>>>>>>>>>>:]
[2024-08-15 20:38:51,531: INFO: main: >>>>>>>> Model training stage started <<<<<<<<:]
[2024-08-15 20:38:51,538: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:38:51,541: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:38:51,548: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:38:51,549: INFO: common: created directory at artifacts:]
[2024-08-15 20:38:51,550: INFO: common: created directory at artifacts/model_trainer:]
[2024-08-15 20:38:51,592: INFO: main: >>>>>>>> Model training stage completed <<<<<<<<:]
[2024-08-15 20:38:51,592: INFO: main: >>>>>>>>>>Model Evaluation stage started <<<<<<<<<<:]
[2024-08-15 20:38:51,597: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:38:51,600: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:38:51,609: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:38:51,611: INFO: common: created directory at artifacts:]
[2024-08-15 20:38:53,129: ERROR: main: [Errno 2] No such file or directory: 'artifacts/model_evaluation/metrics.json':]
Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/main.py", line 60, in <module>
    obj.main()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/pipeline/stage_05_model_evaluation.py", line 18, in main
    model_evaluation.log_into_mlflow()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/components/model_evaluation.py", line 40, in log_into_mlflow
    save_json(path=Path(self.config.metric_file_name), data=scores)
  File "/Users/deepa/.local/pip3/venv/lib/python3.12/site-packages/ensure/main.py", line 835, in __call__
    return self.f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/utils/common.py", line 44, in save_json
    with open(path, "w+") as f:
         ^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: 'artifacts/model_evaluation/metrics.json'
[2024-08-15 20:43:14,593: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 20:43:14,598: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:43:14,611: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:43:14,617: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:43:14,620: INFO: common: created directory at artifacts:]
[2024-08-15 20:43:14,621: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 20:43:15,279: INFO: data_ingestion: artifacts/data_ingestion/data.zip download with following headers Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: F880:70F30:7ACEF6:801944:66BEC6FF
Accept-Ranges: bytes
Date: Fri, 16 Aug 2024 03:43:16 GMT
Via: 1.1 varnish
X-Served-By: cache-pao-kpao1770073-PAO
X-Cache: HIT
X-Cache-Hits: 0
X-Timer: S1723779796.385102,VS0,VE1
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: f76a3dbf056a62addce9f384e24f260aa9bd5820
Expires: Fri, 16 Aug 2024 03:48:16 GMT
Source-Age: 264

:]
[2024-08-15 20:43:15,300: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has completed

**********:]
[2024-08-15 20:43:15,301: INFO: main: starting <<<<<<<<< Data validation Stage >>>>>>>>>>>:]
[2024-08-15 20:43:15,310: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:43:15,312: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:43:15,326: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:43:15,327: INFO: common: created directory at artifacts:]
[2024-08-15 20:43:15,327: INFO: common: created directory at artifacts/data_validation:]
[2024-08-15 20:43:15,420: INFO: main:  <<<<<<<<< Data validation Stage completed >>>>>>>>>>>:]
[2024-08-15 20:43:15,420: INFO: main: <<<<<<<<<< Data Transformation stage started >>>>>>>>>>>>>:]
[2024-08-15 20:43:15,424: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:43:15,426: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:43:15,429: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:43:15,430: INFO: common: created directory at artifacts:]
[2024-08-15 20:43:15,435: INFO: common: created directory at artifacts/data_transformation:]
[2024-08-15 20:43:15,807: INFO: data_transformation: splitting data into train and test data:]
[2024-08-15 20:43:15,807: INFO: data_transformation: (1199, 12):]
[2024-08-15 20:43:15,808: INFO: data_transformation: (400, 12):]
[2024-08-15 20:43:15,809: INFO: main: <<<<<<<<<< Data Transformation stage completed >>>>>>>>>>>>>:]
[2024-08-15 20:43:15,810: INFO: main: >>>>>>>> Model training stage started <<<<<<<<:]
[2024-08-15 20:43:15,824: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:43:15,826: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:43:15,841: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:43:15,842: INFO: common: created directory at artifacts:]
[2024-08-15 20:43:15,843: INFO: common: created directory at artifacts/model_trainer:]
[2024-08-15 20:43:16,105: INFO: main: >>>>>>>> Model training stage completed <<<<<<<<:]
[2024-08-15 20:43:16,105: INFO: main: >>>>>>>>>>Model Evaluation stage started <<<<<<<<<<:]
[2024-08-15 20:43:16,110: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 20:43:16,112: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 20:43:16,119: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 20:43:16,121: INFO: common: created directory at artifacts:]
[2024-08-15 20:43:16,123: INFO: common: created directory at artifacts/model_evaluation:]
[2024-08-15 20:43:17,330: INFO: common: file created at artifacts/model_evaluation/metrics.json:]
[2024-08-15 20:43:30,596: INFO: main: >>>>>>>>>>Model Evaluation stage completed <<<<<<<<<<:]
[2024-08-15 23:05:15,843: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 23:05:15,861: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 23:05:15,867: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 23:05:15,874: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 23:05:15,911: INFO: common: created directory at artifacts:]
[2024-08-15 23:05:15,911: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 23:05:15,912: ERROR: main: cannot access local variable 'filename' where it is not associated with a value:]
Traceback (most recent call last):
  File "/Users/deepa/End-to-end-ML-project/main.py", line 15, in <module>
    obj.main()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/pipeline/stage_01_data_ingestion.py", line 16, in main
    data_ingestion.download_file()
  File "/Users/deepa/End-to-end-ML-project/src/mlProject/components/data_ingestion.py", line 20, in download_file
    logger.info(f"{filename} already exists with file size {get_size(Path(self.config.local_data_file))}")
                   ^^^^^^^^
UnboundLocalError: cannot access local variable 'filename' where it is not associated with a value
[2024-08-15 23:05:46,948: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 23:05:46,958: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 23:05:46,979: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 23:05:46,988: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 23:05:47,011: INFO: common: created directory at artifacts:]
[2024-08-15 23:05:47,012: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 23:05:48,839: INFO: data_ingestion: artifacts/data_ingestion/data.zip download with following headers Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 60D1:142673:895D9B:8F796E:66BEEC3B
Accept-Ranges: bytes
Date: Fri, 16 Aug 2024 06:05:51 GMT
Via: 1.1 varnish
X-Served-By: cache-pao-kpao1770076-PAO
X-Cache: MISS
X-Cache-Hits: 0
X-Timer: S1723788351.870786,VS0,VE155
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: 5d225a5b91fc46762ba6a79c4916bf8d0f2e1009
Expires: Fri, 16 Aug 2024 06:10:51 GMT
Source-Age: 0

:]
[2024-08-15 23:05:48,846: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has completed

**********:]
[2024-08-15 23:05:48,847: INFO: main: starting <<<<<<<<< Data validation Stage >>>>>>>>>>>:]
[2024-08-15 23:05:48,855: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 23:05:48,857: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 23:05:48,865: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 23:05:48,870: INFO: common: created directory at artifacts:]
[2024-08-15 23:05:48,871: INFO: common: created directory at artifacts/data_validation:]
[2024-08-15 23:05:48,910: INFO: main:  <<<<<<<<< Data validation Stage completed >>>>>>>>>>>:]
[2024-08-15 23:05:48,910: INFO: main: <<<<<<<<<< Data Transformation stage started >>>>>>>>>>>>>:]
[2024-08-15 23:05:48,913: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 23:05:48,914: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 23:05:48,921: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 23:05:48,924: INFO: common: created directory at artifacts:]
[2024-08-15 23:05:48,924: INFO: common: created directory at artifacts/data_transformation:]
[2024-08-15 23:05:48,993: INFO: data_transformation: splitting data into train and test data:]
[2024-08-15 23:05:48,993: INFO: data_transformation: (1199, 12):]
[2024-08-15 23:05:48,993: INFO: data_transformation: (400, 12):]
[2024-08-15 23:05:48,993: INFO: main: <<<<<<<<<< Data Transformation stage completed >>>>>>>>>>>>>:]
[2024-08-15 23:05:48,994: INFO: main: >>>>>>>> Model training stage started <<<<<<<<:]
[2024-08-15 23:05:48,997: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 23:05:48,998: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 23:05:49,003: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 23:05:49,004: INFO: common: created directory at artifacts:]
[2024-08-15 23:05:49,004: INFO: common: created directory at artifacts/model_trainer:]
[2024-08-15 23:05:49,056: INFO: main: >>>>>>>> Model training stage completed <<<<<<<<:]
[2024-08-15 23:05:49,057: INFO: main: >>>>>>>>>>Model Evaluation stage started <<<<<<<<<<:]
[2024-08-15 23:05:49,061: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 23:05:49,062: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 23:05:49,068: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 23:05:49,070: INFO: common: created directory at artifacts:]
[2024-08-15 23:05:49,070: INFO: common: created directory at artifacts/model_evaluation:]
[2024-08-15 23:05:50,134: INFO: common: file created at artifacts/model_evaluation/metrics.json:]
[2024-08-15 23:06:02,647: INFO: main: >>>>>>>>>>Model Evaluation stage completed <<<<<<<<<<:]
[2024-08-15 23:07:24,784: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has started

:]
[2024-08-15 23:07:24,793: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 23:07:24,805: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 23:07:24,809: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 23:07:24,810: INFO: common: created directory at artifacts:]
[2024-08-15 23:07:24,811: INFO: common: created directory at artifacts/data_ingestion:]
[2024-08-15 23:07:26,140: INFO: data_ingestion: artifacts/data_ingestion/data.zip download with following headers Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 60D1:142673:895D9B:8F796E:66BEEC3B
Accept-Ranges: bytes
Date: Fri, 16 Aug 2024 06:07:28 GMT
Via: 1.1 varnish
X-Served-By: cache-pao-kpao1770039-PAO
X-Cache: HIT
X-Cache-Hits: 0
X-Timer: S1723788448.196799,VS0,VE1
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: 615f17b8ceaebd39a4c94ff9c0c3bdadb1e080f3
Expires: Fri, 16 Aug 2024 06:12:28 GMT
Source-Age: 97

:]
[2024-08-15 23:07:26,146: INFO: main: stage >>>>>> Data Ingestion stage <<<<<<<< has completed

**********:]
[2024-08-15 23:07:26,146: INFO: main: starting <<<<<<<<< Data validation Stage >>>>>>>>>>>:]
[2024-08-15 23:07:26,162: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 23:07:26,164: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 23:07:26,175: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 23:07:26,177: INFO: common: created directory at artifacts:]
[2024-08-15 23:07:26,178: INFO: common: created directory at artifacts/data_validation:]
[2024-08-15 23:07:26,216: INFO: main:  <<<<<<<<< Data validation Stage completed >>>>>>>>>>>:]
[2024-08-15 23:07:26,220: INFO: main: <<<<<<<<<< Data Transformation stage started >>>>>>>>>>>>>:]
[2024-08-15 23:07:26,229: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 23:07:26,231: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 23:07:26,243: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 23:07:26,244: INFO: common: created directory at artifacts:]
[2024-08-15 23:07:26,244: INFO: common: created directory at artifacts/data_transformation:]
[2024-08-15 23:07:26,323: INFO: data_transformation: splitting data into train and test data:]
[2024-08-15 23:07:26,324: INFO: data_transformation: (1199, 12):]
[2024-08-15 23:07:26,324: INFO: data_transformation: (400, 12):]
[2024-08-15 23:07:26,325: INFO: main: <<<<<<<<<< Data Transformation stage completed >>>>>>>>>>>>>:]
[2024-08-15 23:07:26,325: INFO: main: >>>>>>>> Model training stage started <<<<<<<<:]
[2024-08-15 23:07:26,346: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 23:07:26,347: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 23:07:26,355: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 23:07:26,356: INFO: common: created directory at artifacts:]
[2024-08-15 23:07:26,356: INFO: common: created directory at artifacts/model_trainer:]
[2024-08-15 23:07:26,392: INFO: main: >>>>>>>> Model training stage completed <<<<<<<<:]
[2024-08-15 23:07:26,393: INFO: main: >>>>>>>>>>Model Evaluation stage started <<<<<<<<<<:]
[2024-08-15 23:07:26,397: INFO: common: yaml file config/config.yaml loaded successfully:]
[2024-08-15 23:07:26,399: INFO: common: yaml file params.yaml loaded successfully:]
[2024-08-15 23:07:26,407: INFO: common: yaml file schema.yaml loaded successfully:]
[2024-08-15 23:07:26,408: INFO: common: created directory at artifacts:]
[2024-08-15 23:07:26,408: INFO: common: created directory at artifacts/model_evaluation:]
[2024-08-15 23:07:27,105: INFO: common: file created at artifacts/model_evaluation/metrics.json:]
[2024-08-15 23:07:39,689: INFO: main: >>>>>>>>>>Model Evaluation stage completed <<<<<<<<<<:]
